<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cogeqc">
<title>Assessing genome assembly and annotation quality • cogeqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.5/font.css" rel="stylesheet">
<link href="../deps/Source_Code_Pro-0.4.5/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Assessing genome assembly and annotation quality">
<meta property="og:description" content="cogeqc">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-#0092ac"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cogeqc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignette_01_assessing_genome_assembly.html">Assessing genome assembly and annotation quality</a>
    <a class="dropdown-item" href="../articles/vignette_02_assessing_orthogroup_inference.html">Assessing orthogroup inference</a>
    <a class="dropdown-item" href="../articles/vignette_03_assessing_synteny.html">Assessing synteny identification</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/almeidasilvaf/cogeqc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Assessing genome assembly and annotation quality</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent,
Belgium<br><h4 data-toc-skip class="author">Yves
Van de Peer</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent, Belgium<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/cogeqc/blob/HEAD/vignettes/vignette_01_assessing_genome_assembly.Rmd" class="external-link"><code>vignettes/vignette_01_assessing_genome_assembly.Rmd</code></a></small>
      <div class="d-none name"><code>vignette_01_assessing_genome_assembly.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>When working on your own genome project or when using publicly
available genomes for comparative analyses, it is critical to assess the
quality of your data. Over the past years, several tools have been
developed and several metrics have been proposed to assess the quality
of a genome assembly and annotation. <code>cogeqc</code> helps users
interpret their genome assembly statistics by comparing them with
statistics on publicly available genomes on the NCBI. Additionally,
<code>cogeqc</code> also provides an interface to BUSCO <span class="citation">(Simão et al. 2015)</span>, a popular tool to assess
gene space completeness. Graphical functions are available to make
publication-ready plots that summarize the results of quality
control.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <code>cogeqc</code> from Bioconductor with the
following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"cogeqc"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package after installation</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/cogeqc" class="external-link">cogeqc</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="assessing-genome-assembly-quality-statistics-in-a-context">Assessing genome assembly quality: statistics in a context<a class="anchor" aria-label="anchor" href="#assessing-genome-assembly-quality-statistics-in-a-context"></a>
</h2>
<p>When analyzing and interpreting genome assembly statistics, it is
often useful to place your stats in a context by comparing them with
stats from genomes of closely-related or even the same species.
<code>cogeqc</code> provides users with an interface to the NCBI
Datasets API, which can be used to retrieve summary stats for genomes on
NCBI. In this section, we will guide you on how to retrieve such
information and use it as a reference to interpret your data.</p>
<div class="section level3">
<h3 id="obtaining-assembly-statistics-for-ncbi-genomes">Obtaining assembly statistics for NCBI genomes<a class="anchor" aria-label="anchor" href="#obtaining-assembly-statistics-for-ncbi-genomes"></a>
</h3>
<p>To obtain a data frame of summary statistics for NCBI genomes of a
particular taxon, you will use the function
<code><a href="../reference/get_genome_stats.html">get_genome_stats()</a></code>. In the <em>taxon</em> parameter, you
must specify the taxon from which data will be extracted. This can be
done either by passing a character scalar with taxon name or by passing
a numeric scalar with NCBI Taxonomy ID. For example, the code below
demonstrates two ways of extracting stats on maize (<em>Zea mays</em>)
genomes on NCBI:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example 1: get stats for all maize genomes using taxon name</span></span>
<span><span class="va">maize_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genome_stats.html">get_genome_stats</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">"Zea mays"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">maize_stats</span><span class="op">)</span></span>
<span><span class="co">#&gt;         accession  source species_taxid         species_name</span></span>
<span><span class="co">#&gt; 1 GCA_000005005.6 GENBANK          4577             Zea mays</span></span>
<span><span class="co">#&gt; 2 GCA_000223545.1 GENBANK          4577             Zea mays</span></span>
<span><span class="co">#&gt; 3 GCA_000275765.1 GENBANK          4577             Zea mays</span></span>
<span><span class="co">#&gt; 4 GCA_001644905.2 GENBANK        381124 Zea mays subsp. mays</span></span>
<span><span class="co">#&gt; 5 GCA_001984235.2 GENBANK        381124 Zea mays subsp. mays</span></span>
<span><span class="co">#&gt; 6 GCA_001990705.1 GENBANK        381124 Zea mays subsp. mays</span></span>
<span><span class="co">#&gt;   species_common_name species_ecotype species_strain species_isolate</span></span>
<span><span class="co">#&gt; 1               maize            &lt;NA&gt;             NA            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2               maize            &lt;NA&gt;             NA            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3               maize            &lt;NA&gt;             NA            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4               maize            &lt;NA&gt;             NA            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5               maize            &lt;NA&gt;             NA            &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6               maize            &lt;NA&gt;             NA            &lt;NA&gt;</span></span>
<span><span class="co">#&gt;               species_cultivar assembly_level assembly_status</span></span>
<span><span class="co">#&gt; 1                          B73     Chromosome         current</span></span>
<span><span class="co">#&gt; 2 Palomero Toluqueno EDMX-2231       Scaffold         current</span></span>
<span><span class="co">#&gt; 3                          B73         Contig         current</span></span>
<span><span class="co">#&gt; 4                          W22     Chromosome         current</span></span>
<span><span class="co">#&gt; 5                          EP1     Chromosome         current</span></span>
<span><span class="co">#&gt; 6                           F7     Chromosome         current</span></span>
<span><span class="co">#&gt;                 assembly_name assembly_type submission_date</span></span>
<span><span class="co">#&gt; 1               B73 RefGen_v4       haploid      2017-02-07</span></span>
<span><span class="co">#&gt; 2     ZeaMays_PT_EDMX2233_1.0       haploid      2011-08-16</span></span>
<span><span class="co">#&gt; 3            ZmaysB73_wgs_1.0       haploid      2012-07-03</span></span>
<span><span class="co">#&gt; 4 Zm-W22-REFERENCE-NRGENE-2.0       haploid      2017-02-02</span></span>
<span><span class="co">#&gt; 5    Zm-EP1-REFERENCE-TUM-1.0       haploid      2017-02-09</span></span>
<span><span class="co">#&gt; 6     Zm-F7-REFERENCE-TUM-1.0       haploid      2017-02-09</span></span>
<span><span class="co">#&gt;                                                                              submitter</span></span>
<span><span class="co">#&gt; 1                                                                        maizesequence</span></span>
<span><span class="co">#&gt; 2 Laboratorio Nacional de Genómica para la Biodiversidad (LANGEBIO) CINVESTAV Irapuato</span></span>
<span><span class="co">#&gt; 3                                                                        maizesequence</span></span>
<span><span class="co">#&gt; 4                                                            W22 Sequencing Consortium</span></span>
<span><span class="co">#&gt; 5                                       Technical University of Munich, Plant Breeding</span></span>
<span><span class="co">#&gt; 6                                       Technical University of Munich, Plant Breeding</span></span>
<span><span class="co">#&gt;            sequencing_technology atypical refseq_category chromosome_count</span></span>
<span><span class="co">#&gt; 1                         PacBio    FALSE            &lt;NA&gt;               10</span></span>
<span><span class="co">#&gt; 2 454 GS20; 454 Titanium; Sanger     TRUE            &lt;NA&gt;               NA</span></span>
<span><span class="co">#&gt; 3                     454 GS FLX     TRUE            &lt;NA&gt;               NA</span></span>
<span><span class="co">#&gt; 4   Illumina HiSeq; 10x Genomics    FALSE            &lt;NA&gt;               10</span></span>
<span><span class="co">#&gt; 5                       Illumina    FALSE            &lt;NA&gt;               10</span></span>
<span><span class="co">#&gt; 6                       Illumina    FALSE            &lt;NA&gt;               10</span></span>
<span><span class="co">#&gt;   sequence_length ungapped_length contig_count contig_N50 contig_L50</span></span>
<span><span class="co">#&gt; 1      2134373047      2103640169         2787    1279870        506</span></span>
<span><span class="co">#&gt; 2       177051422       177051324       196705       1129      44576</span></span>
<span><span class="co">#&gt; 3         1335069         1335069         1844        949        434</span></span>
<span><span class="co">#&gt; 4      2133880228      2093255169        68134      72426       8765</span></span>
<span><span class="co">#&gt; 5      2455259639      2429778437       130125      82504       8782</span></span>
<span><span class="co">#&gt; 6      2392801755      2367184004       117311      96966       7318</span></span>
<span><span class="co">#&gt;   scaffold_count scaffold_N50 scaffold_L50 GC_percent annotation_provider</span></span>
<span><span class="co">#&gt; 1            596     10679170           62       46.5       maizesequence</span></span>
<span><span class="co">#&gt; 2         196697         1129        44576       45.5                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3             NA           NA           NA       45.0                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4            191    222590201            5       46.5                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5          60567    249966684            5       46.5                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6          62610    237276974            5       46.5                &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   annotation_release_date gene_count_total gene_count_coding</span></span>
<span><span class="co">#&gt; 1              2017-02-07            39320             39320</span></span>
<span><span class="co">#&gt; 2                    &lt;NA&gt;               NA                NA</span></span>
<span><span class="co">#&gt; 3                    &lt;NA&gt;               NA                NA</span></span>
<span><span class="co">#&gt; 4                    &lt;NA&gt;               NA                NA</span></span>
<span><span class="co">#&gt; 5                    &lt;NA&gt;               NA                NA</span></span>
<span><span class="co">#&gt; 6                    &lt;NA&gt;               NA                NA</span></span>
<span><span class="co">#&gt;   gene_count_noncoding gene_count_pseudogene gene_count_other CC_ratio</span></span>
<span><span class="co">#&gt; 1                   NA                    NA               NA    278.7</span></span>
<span><span class="co">#&gt; 2                   NA                    NA               NA       NA</span></span>
<span><span class="co">#&gt; 3                   NA                    NA               NA       NA</span></span>
<span><span class="co">#&gt; 4                   NA                    NA               NA   6813.4</span></span>
<span><span class="co">#&gt; 5                   NA                    NA               NA  13012.5</span></span>
<span><span class="co">#&gt; 6                   NA                    NA               NA  11731.1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">maize_stats</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'data.frame':    94 obs. of  36 variables:</span></span>
<span><span class="co">#&gt;  $ accession              : chr  "GCA_000005005.6" "GCA_000223545.1" "GCA_000275765.1" "GCA_001644905.2" ...</span></span>
<span><span class="co">#&gt;  $ source                 : chr  "GENBANK" "GENBANK" "GENBANK" "GENBANK" ...</span></span>
<span><span class="co">#&gt;  $ species_taxid          : int  4577 4577 4577 381124 381124 381124 381124 381124 4579 381124 ...</span></span>
<span><span class="co">#&gt;  $ species_name           : chr  "Zea mays" "Zea mays" "Zea mays" "Zea mays subsp. mays" ...</span></span>
<span><span class="co">#&gt;  $ species_common_name    : chr  "maize" "maize" "maize" "maize" ...</span></span>
<span><span class="co">#&gt;  $ species_ecotype        : chr  NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ species_strain         : logi  NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ species_isolate        : chr  NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ species_cultivar       : chr  "B73" "Palomero Toluqueno EDMX-2231" "B73" "W22" ...</span></span>
<span><span class="co">#&gt;  $ assembly_level         : Factor w/ 4 levels "Complete","Chromosome",..: 2 3 4 2 2 2 2 2 3 3 ...</span></span>
<span><span class="co">#&gt;  $ assembly_status        : chr  "current" "current" "current" "current" ...</span></span>
<span><span class="co">#&gt;  $ assembly_name          : chr  "B73 RefGen_v4" "ZeaMays_PT_EDMX2233_1.0" "ZmaysB73_wgs_1.0" "Zm-W22-REFERENCE-NRGENE-2.0" ...</span></span>
<span><span class="co">#&gt;  $ assembly_type          : chr  "haploid" "haploid" "haploid" "haploid" ...</span></span>
<span><span class="co">#&gt;  $ submission_date        : chr  "2017-02-07" "2011-08-16" "2012-07-03" "2017-02-02" ...</span></span>
<span><span class="co">#&gt;  $ submitter              : chr  "maizesequence" "Laboratorio Nacional de Genómica para la Biodiversidad (LANGEBIO) CINVESTAV Irapuato" "maizesequence" "W22 Sequencing Consortium" ...</span></span>
<span><span class="co">#&gt;  $ sequencing_technology  : chr  "PacBio" "454 GS20; 454 Titanium; Sanger" "454 GS FLX" "Illumina HiSeq; 10x Genomics" ...</span></span>
<span><span class="co">#&gt;  $ atypical               : logi  FALSE TRUE TRUE FALSE FALSE FALSE ...</span></span>
<span><span class="co">#&gt;  $ refseq_category        : chr  NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ chromosome_count       : int  10 NA NA 10 10 10 10 10 NA NA ...</span></span>
<span><span class="co">#&gt;  $ sequence_length        : num  2.13e+09 1.77e+08 1.34e+06 2.13e+09 2.46e+09 ...</span></span>
<span><span class="co">#&gt;  $ ungapped_length        : num  2.10e+09 1.77e+08 1.34e+06 2.09e+09 2.43e+09 ...</span></span>
<span><span class="co">#&gt;  $ contig_count           : int  2787 196705 1844 68134 130125 117311 405730 182074 140894 102656 ...</span></span>
<span><span class="co">#&gt;  $ contig_N50             : int  1279870 1129 949 72426 82504 96966 10873 39757 31764 71122 ...</span></span>
<span><span class="co">#&gt;  $ contig_L50             : int  506 44576 434 8765 8782 7318 42688 16302 7912 8373 ...</span></span>
<span><span class="co">#&gt;  $ scaffold_count         : int  596 196697 NA 191 60567 62610 43299 3538 107418 48268 ...</span></span>
<span><span class="co">#&gt;  $ scaffold_N50           : int  10679170 1129 NA 222590201 249966684 237276974 215148664 2707071 107689 2995073 ...</span></span>
<span><span class="co">#&gt;  $ scaffold_L50           : int  62 44576 NA 5 5 5 5 233 499 168 ...</span></span>
<span><span class="co">#&gt;  $ GC_percent             : num  46.5 45.5 45 46.5 46.5 46.5 46 46.5 45.5 46 ...</span></span>
<span><span class="co">#&gt;  $ annotation_provider    : chr  "maizesequence" NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ annotation_release_date: chr  "2017-02-07" NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ gene_count_total       : int  39320 NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ gene_count_coding      : int  39320 NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ gene_count_noncoding   : int  NA NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ gene_count_pseudogene  : int  NA NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ gene_count_other       : int  NA NA NA NA NA NA NA NA NA NA ...</span></span>
<span><span class="co">#&gt;  $ CC_ratio               : num  279 NA NA 6813 13012 ...</span></span>
<span></span>
<span><span class="co"># Example 2: get stats for all maize genomes using NCBI Taxonomy ID</span></span>
<span><span class="va">maize_stats2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genome_stats.html">get_genome_stats</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="fl">4577</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Checking if objects are the same</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">maize_stats</span>, <span class="va">maize_stats2</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>As you can see, there are 94 maize genomes on the NCBI. You can also
include filters in your searches by passing a list of key-value pairs
with keys in list names and values in elements. For instance, to obtain
only <strong>chromosome-scale</strong> and <strong>annotated</strong>
maize genomes, you would run:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get chromosome-scale maize genomes with annotation</span></span>
<span><span class="co">## Create list of filters</span></span>
<span><span class="va">filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    filters.has_annotation <span class="op">=</span> <span class="st">"true"</span>,</span>
<span>    filters.assembly_level <span class="op">=</span> <span class="st">"chromosome"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">filt</span></span>
<span><span class="co">#&gt; $filters.has_annotation</span></span>
<span><span class="co">#&gt; [1] "true"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $filters.assembly_level</span></span>
<span><span class="co">#&gt; [1] "chromosome"</span></span>
<span></span>
<span><span class="co">## Obtain data</span></span>
<span><span class="va">filtered_maize_genomes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genome_stats.html">get_genome_stats</a></span><span class="op">(</span>taxon <span class="op">=</span> <span class="st">"Zea mays"</span>, filters <span class="op">=</span> <span class="va">filt</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">filtered_maize_genomes</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  4 36</span></span></code></pre></div>
<p>For a full list of filtering parameters and possible arguments, see
the <a href="https://www.ncbi.nlm.nih.gov/datasets/docs/v2/reference-docs/rest-api/#get-/genome/taxon/-taxons-/dataset_report" class="external-link">API
documentation</a>.</p>
</div>
<div class="section level3">
<h3 id="comparing-custom-stats-with-ncbi-stats">Comparing custom stats with NCBI stats<a class="anchor" aria-label="anchor" href="#comparing-custom-stats-with-ncbi-stats"></a>
</h3>
<p>Now, suppose you sequenced a genome, obtained assembly and annotation
stats, and want to compare them to NCBI genomes to identify potential
issues. Examples of situations you may encounter include:</p>
<ul>
<li><p>The genome you assembled is huge and you think there might be a
problem with your assembly.</p></li>
<li><p>Your gene annotation pipeline predicted <em>n</em> genes, but you
are not sure if this number is reasonable compared to other assemblies
of the same species or closely-related species.</p></li>
</ul>
<p>To compare user-defined summary stats with NCBI stats, you will use
the function <code><a href="../reference/compare_genome_stats.html">compare_genome_stats()</a></code>. This function will
include the values you observed for each statistic into a distribution
(based on NCBI stats) and return the percentile and rank of your
observed values in each distribution.</p>
<p>As an example, let’s go back to our maize stats we obtained in the
previous section. Suppose you sequenced a new maize genome and observed
the following values:</p>
<ol style="list-style-type: decimal">
<li>Genome size = 2.4 Gb</li>
<li>Number of genes = 50,000</li>
<li>CC ratio = 2 <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;strong&gt;Note:&lt;/strong&gt; The CC ratio is the ratio of the
number of contigs to the number of chromosome pairs, and it has been
proposed in &lt;span class="citation"&gt;Wang and Wang (2022)&lt;/span&gt; as a
measurement of contiguity that compensates for the flaws of N50 and
allows cross-species comparisons.&lt;/p&gt;'><sup>1</sup></a>
</li>
</ol>
<p>To compare your observed values with those for publicly available
maize genomes, you need to store them in a data frame. The column
<strong>accession</strong> is mandatory, and any other column will be
matched against columns in the data frame obtained with
<code><a href="../reference/get_genome_stats.html">get_genome_stats()</a></code>. Thus, make sure column names in your
data frame match column names in the reference data frame. Then, you can
compare both data frames as below:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check column names in the data frame of stats for maize genomes on the NCBI</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">maize_stats</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "accession"               "source"                 </span></span>
<span><span class="co">#&gt;  [3] "species_taxid"           "species_name"           </span></span>
<span><span class="co">#&gt;  [5] "species_common_name"     "species_ecotype"        </span></span>
<span><span class="co">#&gt;  [7] "species_strain"          "species_isolate"        </span></span>
<span><span class="co">#&gt;  [9] "species_cultivar"        "assembly_level"         </span></span>
<span><span class="co">#&gt; [11] "assembly_status"         "assembly_name"          </span></span>
<span><span class="co">#&gt; [13] "assembly_type"           "submission_date"        </span></span>
<span><span class="co">#&gt; [15] "submitter"               "sequencing_technology"  </span></span>
<span><span class="co">#&gt; [17] "atypical"                "refseq_category"        </span></span>
<span><span class="co">#&gt; [19] "chromosome_count"        "sequence_length"        </span></span>
<span><span class="co">#&gt; [21] "ungapped_length"         "contig_count"           </span></span>
<span><span class="co">#&gt; [23] "contig_N50"              "contig_L50"             </span></span>
<span><span class="co">#&gt; [25] "scaffold_count"          "scaffold_N50"           </span></span>
<span><span class="co">#&gt; [27] "scaffold_L50"            "GC_percent"             </span></span>
<span><span class="co">#&gt; [29] "annotation_provider"     "annotation_release_date"</span></span>
<span><span class="co">#&gt; [31] "gene_count_total"        "gene_count_coding"      </span></span>
<span><span class="co">#&gt; [33] "gene_count_noncoding"    "gene_count_pseudogene"  </span></span>
<span><span class="co">#&gt; [35] "gene_count_other"        "CC_ratio"</span></span>
<span></span>
<span><span class="co"># Create a simulated data frame of stats for a maize genome</span></span>
<span><span class="va">my_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    accession <span class="op">=</span> <span class="st">"my_lovely_maize"</span>,</span>
<span>    sequence_length <span class="op">=</span> <span class="fl">2.4</span> <span class="op">*</span> <span class="fl">1e9</span>,</span>
<span>    gene_count_total <span class="op">=</span> <span class="fl">50000</span>,</span>
<span>    CC_ratio <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare stats</span></span>
<span><span class="fu"><a href="../reference/compare_genome_stats.html">compare_genome_stats</a></span><span class="op">(</span>ncbi_stats <span class="op">=</span> <span class="va">maize_stats</span>, user_stats <span class="op">=</span> <span class="va">my_stats</span><span class="op">)</span></span>
<span><span class="co">#&gt;         accession         variable percentile rank</span></span>
<span><span class="co">#&gt; 1 my_lovely_maize  sequence_length 0.97894737    3</span></span>
<span><span class="co">#&gt; 2 my_lovely_maize gene_count_total 1.00000000    1</span></span>
<span><span class="co">#&gt; 3 my_lovely_maize         CC_ratio 0.02941176    2</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualizing-summary-assembly-statistics">Visualizing summary assembly statistics<a class="anchor" aria-label="anchor" href="#visualizing-summary-assembly-statistics"></a>
</h3>
<p>To have a visual representation of the summary stats obtained with
<code><a href="../reference/get_genome_stats.html">get_genome_stats()</a></code>, you will use the function
<code><a href="../reference/plot_genome_stats.html">plot_genome_stats()</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summarize genome stats in a plot</span></span>
<span><span class="fu"><a href="../reference/plot_genome_stats.html">plot_genome_stats</a></span><span class="op">(</span>ncbi_stats <span class="op">=</span> <span class="va">maize_stats</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_01_assessing_genome_assembly_files/figure-html/plot_genome_stats-1.png" width="1000"></p>
<p>Finally, you can pass your data frame of observed stats to highlight
your values (as red points) in the distributions.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_genome_stats.html">plot_genome_stats</a></span><span class="op">(</span>ncbi_stats <span class="op">=</span> <span class="va">maize_stats</span>, user_stats <span class="op">=</span> <span class="va">my_stats</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_01_assessing_genome_assembly_files/figure-html/plot_genome_stats_with_user_stats-1.png" width="1000"></p>
</div>
</div>
<div class="section level2">
<h2 id="assessing-gene-space-completeness-with-busco">Assessing gene space completeness with BUSCO<a class="anchor" aria-label="anchor" href="#assessing-gene-space-completeness-with-busco"></a>
</h2>
<p>One of the most common metrics to assess gene space completeness is
BUSCO (best universal single-copy orthologs) <span class="citation">(Simão et al. 2015)</span>. <code>cogeqc</code> allows
users to run BUSCO from an R session and visualize results graphically.
BUSCO summary statistics will help you assess which assemblies have high
quality based on the percentage of complete BUSCOs.</p>
<div class="section level3">
<h3 id="running-busco">Running BUSCO<a class="anchor" aria-label="anchor" href="#running-busco"></a>
</h3>
<p>To run BUSCO from R, you will use the function
<code><a href="../reference/run_busco.html">run_busco()</a></code><a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;strong&gt;Note:&lt;/strong&gt; You must have BUSCO installed and
in your PATH to use &lt;code&gt;&lt;a href="../reference/run_busco.html"&gt;run_busco()&lt;/a&gt;&lt;/code&gt;. You can check if BUSCO is
installed by running &lt;code&gt;&lt;a href="../reference/busco_is_installed.html"&gt;busco_is_installed()&lt;/a&gt;&lt;/code&gt;. If you don’t
have it already, you can manually install it or use a conda virtual
environment with the Bioconductor package &lt;code&gt;Herper&lt;/code&gt; &lt;span class="citation"&gt;(Paul, Carroll, and Barrows 2021)&lt;/span&gt;.&lt;/p&gt;'><sup>2</sup></a>. Here, we will use an example FASTA file
containing the first 1,000 lines of the <em>Herbaspirilllum seropedicae
SmR1</em> genome (GCA_000143225), which was downloaded from Ensembl
Bacteria. We will run BUSCO using <em>burkholderiales_odb10</em> as the
lineage dataset. To view all available datasets, run
<code><a href="../reference/list_busco_datasets.html">list_busco_datasets()</a></code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Path to FASTA file</span></span>
<span><span class="va">sequence</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Hse_subset.fa"</span>, package <span class="op">=</span> <span class="st">"cogeqc"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Path to directory where BUSCO datasets will be stored</span></span>
<span><span class="va">download_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"/datasets"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run BUSCO if it is installed</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="../reference/busco_is_installed.html">busco_is_installed</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/run_busco.html">run_busco</a></span><span class="op">(</span><span class="va">sequence</span>, outlabel <span class="op">=</span> <span class="st">"Hse"</span>, mode <span class="op">=</span> <span class="st">"genome"</span>,</span>
<span>            lineage <span class="op">=</span> <span class="st">"burkholderiales_odb10"</span>,</span>
<span>            outpath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, download_path <span class="op">=</span> <span class="va">download_path</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>The output will be stored in the directory specified in
<em>outpath</em>. You can read and parse BUSCO’s output with the
function <code><a href="../reference/read_busco.html">read_busco()</a></code>. For example, let’s read the output
of a BUSCO run using the genome of the green algae <em>Ostreococcus
tauri</em>. The output directory is <code>/extdata</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Path to output directory</span></span>
<span><span class="va">output_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"cogeqc"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">busco_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_busco.html">read_busco</a></span><span class="op">(</span><span class="va">output_dir</span><span class="op">)</span></span>
<span><span class="va">busco_summary</span></span>
<span><span class="co">#&gt;                Class Frequency           Lineage</span></span>
<span><span class="co">#&gt; 1        Complete_SC      1412 chlorophyta_odb10</span></span>
<span><span class="co">#&gt; 2 Complete_duplicate         4 chlorophyta_odb10</span></span>
<span><span class="co">#&gt; 3         Fragmented        35 chlorophyta_odb10</span></span>
<span><span class="co">#&gt; 4            Missing        68 chlorophyta_odb10</span></span></code></pre></div>
<p>This is an example output for a BUSCO run with a single FASTA file.
You can also specify a directory containing multiple FASTA files in the
<em>sequence</em> argument of <code><a href="../reference/run_busco.html">run_busco()</a></code>. This way, BUSCO
will be run in batch mode. Let’s see what the output of BUSCO in batch
mode looks like:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">batch_summary</span><span class="op">)</span></span>
<span><span class="va">batch_summary</span></span>
<span><span class="co">#&gt;                Class Frequency               Lineage   File</span></span>
<span><span class="co">#&gt; 1        Complete_SC      98.5 burkholderiales_odb10 Hse.fa</span></span>
<span><span class="co">#&gt; 2        Complete_SC      98.8 burkholderiales_odb10 Hru.fa</span></span>
<span><span class="co">#&gt; 3 Complete_duplicate       0.7 burkholderiales_odb10 Hse.fa</span></span>
<span><span class="co">#&gt; 4 Complete_duplicate       0.7 burkholderiales_odb10 Hru.fa</span></span>
<span><span class="co">#&gt; 5         Fragmented       0.4 burkholderiales_odb10 Hse.fa</span></span>
<span><span class="co">#&gt; 6         Fragmented       0.3 burkholderiales_odb10 Hru.fa</span></span>
<span><span class="co">#&gt; 7            Missing       0.4 burkholderiales_odb10 Hse.fa</span></span>
<span><span class="co">#&gt; 8            Missing       0.2 burkholderiales_odb10 Hru.fa</span></span></code></pre></div>
<p>The only difference between this data frame and the previous one is
the column <strong>File</strong>, which contains information on the
FASTA file. The example dataset <code>batch_summary</code> contains the
output of <code><a href="../reference/run_busco.html">run_busco()</a></code> using a directory containing two
genomes (<em>Herbaspirillum seropedicae SmR1</em> and <em>Herbaspirillum
rubrisubalbicans M1</em>) as parameter to the <em>sequence</em>
argument.</p>
</div>
<div class="section level3">
<h3 id="visualizing-busco-summary-statistics">Visualizing BUSCO summary statistics<a class="anchor" aria-label="anchor" href="#visualizing-busco-summary-statistics"></a>
</h3>
<p>After using <code><a href="../reference/run_busco.html">run_busco()</a></code> and parsing its output with
<code><a href="../reference/read_busco.html">read_busco()</a></code>, users can visualize summary statistics with
<code><a href="../reference/plot_busco.html">plot_busco()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Single FASTA file - Ostreococcus tauri</span></span>
<span><span class="fu"><a href="../reference/plot_busco.html">plot_busco</a></span><span class="op">(</span><span class="va">busco_summary</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_01_assessing_genome_assembly_files/figure-html/plot_busco-1.png" width="100%"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Batch mode - Herbaspirillum seropedicae and H. rubrisubalbicans</span></span>
<span><span class="fu"><a href="../reference/plot_busco.html">plot_busco</a></span><span class="op">(</span><span class="va">batch_summary</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_01_assessing_genome_assembly_files/figure-html/plot_busco-2.png" width="100%"></p>
<p>We usually consider genomes with &gt;90% of complete BUSCOs as having
high quality. Thus, we can conclude that the three genomes analyzed here
are high-quality genomes.</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-03-14</span></span>
<span><span class="co">#&gt;  pandoc   2.18 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package         </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  ape                5.7       <span style="color: #555555;">2023-02-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  aplot              0.1.10    <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  beeswarm           0.4.0     <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics       0.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager        1.30.20   <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocStyle        * 2.24.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings         2.64.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bitops             1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bookdown           0.33      <span style="color: #555555;">2023-03-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bslib              0.4.2     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cachem             1.0.7     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cli                3.6.0     <span style="color: #555555;">2023-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cogeqc           * 1.3.1     <span style="color: #555555;">2023-03-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  colorspace         2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  crayon             1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  desc               1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  digest             0.6.31    <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  dplyr              1.1.0     <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  evaluate           0.20      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fansi              1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  farver             2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fastmap            1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fs                 1.6.1     <span style="color: #555555;">2023-02-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  generics           0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb       1.32.4    <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData   1.2.8     <span style="color: #555555;">2022-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm         0.7.1     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggfun              0.0.9     <span style="color: #555555;">2022-11-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2            3.4.1     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggplotify          0.1.0     <span style="color: #555555;">2021-09-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggtree             3.4.4     <span style="color: #555555;">2022-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  glue               1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gridGraphics       0.5-1     <span style="color: #555555;">2020-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gtable             0.3.1     <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  highr              0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmltools          0.5.4     <span style="color: #555555;">2022-12-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  igraph             1.4.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  IRanges            2.30.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  jquerylib          0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite           1.8.4     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  knitr              1.42      <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  labeling           0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lattice            0.20-45   <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  lazyeval           0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle          1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magrittr           2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  memoise            2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  munsell            0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  nlme               3.1-162   <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  patchwork          1.1.2     <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pillar             1.8.1     <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig          2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown            2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plyr               1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  purrr              1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R6                 2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ragg               1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp               1.0.10    <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RCurl              1.98-1.10 <span style="color: #555555;">2023-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  reshape2           1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rlang              1.0.6     <span style="color: #555555;">2022-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown          2.20      <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot          2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  S4Vectors          0.34.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass               0.4.5     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  scales             1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo        1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringi            1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringr            1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  systemfonts        1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  textshaping        0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tibble             3.2.0     <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyr              1.3.0     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect         1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidytree           0.4.2     <span style="color: #555555;">2022-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  treeio             1.20.2    <span style="color: #555555;">2022-08-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  utf8               1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vctrs              0.5.2     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vipor              0.4.5     <span style="color: #555555;">2017-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  withr              2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xfun               0.37      <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XVector            0.36.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml               2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  yulab.utils        0.0.6     <span style="color: #555555;">2022-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc           1.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-herper" class="csl-entry">
Paul, Matt, Thomas Carroll, and Doug Barrows. 2021. <em>Herper: The
Herper Package Is a Simple Toolset to Install and Manage Conda Packages
and Environments from r</em>. <a href="https://github.com/RockefellerUniversity/Herper" class="external-link">https://github.com/RockefellerUniversity/Herper</a>.
</div>
<div id="ref-simao2015busco" class="csl-entry">
Simão, Felipe A, Robert M Waterhouse, Panagiotis Ioannidis, Evgenia V
Kriventseva, and Evgeny M Zdobnov. 2015. <span>“BUSCO: Assessing Genome
Assembly and Annotation Completeness with Single-Copy Orthologs.”</span>
<em>Bioinformatics</em> 31 (19): 3210–12.
</div>
<div id="ref-wang2022proposed" class="csl-entry">
Wang, Peng, and Fei Wang. 2022. <span>“A Proposed Metric Set for
Evaluation of Genome Assembly Quality.”</span> <em>Trends in
Genetics</em>.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
