<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Assessing orthogroup inference • cogeqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Assessing orthogroup inference">
<meta property="og:description" content="cogeqc">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cogeqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette_01_assessing_genome_assembly.html">Assessing genome assembly quality</a>
    </li>
    <li>
      <a href="../articles/vignette_02_assessing_orthogroup_inference.html">Assessing orthogroup inference</a>
    </li>
    <li>
      <a href="../articles/vignette_03_assessing_synteny.html">Assessing synteny identification</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/almeidasilvaf/cogeqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Assessing orthogroup inference</h1>
                        <h4 data-toc-skip class="author">Fabricio Almeida-Silva</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent, Belgium<br><h4 data-toc-skip class="author">Yves Van de Peer</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent, Belgium<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/cogeqc/blob/HEAD/vignettes/vignette_02_assessing_orthogroup_inference.Rmd" class="external-link"><code>vignettes/vignette_02_assessing_orthogroup_inference.Rmd</code></a></small>
      <div class="hidden name"><code>vignette_02_assessing_orthogroup_inference.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The identification of groups of homologous genes within and across species is a powerful tool for evolutionary genomics. The most widely used tools to identify orthogroups (i.e., groups of orthologous genes) are OrthoFinder <span class="citation">(Emms and Kelly 2019)</span> and OrthoMCL <span class="citation">(Li, Stoeckert, and Roos 2003)</span>. However, these tools generate different results depending on the parameters used, such as <em>mcl inflation parameter</em>, <em>E-value</em>, <em>maximum number of hits</em>, and others. Here, we propose a protein domain-aware assessment of orthogroup inference. The goal is to maximize the percentage of shared protein domains for genes in the same orthogroup.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"cogeqc"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load package after installation</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/cogeqc" class="external-link">cogeqc</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>Here, we will use orthogroups from the PLAZA 5.0 database <span class="citation">(Van Bel et al. 2021)</span>, inferred with OrthoFinder <span class="citation">(Emms and Kelly 2019)</span>. For the purpose of demonstration, the complete dataset was filtered to only keep orthogroups for the Brassicaceae species <em>Arabidopsis thaliana</em> and <em>Brassica oleraceae</em>. Interpro domain annotations were also retrieved from PLAZA 5.0.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Orthogroups for Arabidopsis thaliana and Brassica oleraceae</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">og</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">og</span><span class="op">)</span>
<span class="co">#&gt;     Orthogroup Species      Gene</span>
<span class="co">#&gt; 1 HOM05D000001     Ath AT1G02310</span>
<span class="co">#&gt; 2 HOM05D000001     Ath AT1G03510</span>
<span class="co">#&gt; 3 HOM05D000001     Ath AT1G03540</span>
<span class="co">#&gt; 4 HOM05D000001     Ath AT1G04020</span>
<span class="co">#&gt; 5 HOM05D000001     Ath AT1G04840</span>
<span class="co">#&gt; 6 HOM05D000001     Ath AT1G05750</span>

<span class="co"># Interpro domain annotations</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">interpro_ath</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">interpro_bol</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">interpro_ath</span><span class="op">)</span>
<span class="co">#&gt;        Gene Annotation</span>
<span class="co">#&gt; 1 AT1G01010  IPR036093</span>
<span class="co">#&gt; 2 AT1G01010  IPR003441</span>
<span class="co">#&gt; 3 AT1G01010  IPR036093</span>
<span class="co">#&gt; 4 AT1G01020  IPR007290</span>
<span class="co">#&gt; 5 AT1G01020  IPR007290</span>
<span class="co">#&gt; 6 AT1G01030  IPR003340</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">interpro_bol</span><span class="op">)</span>
<span class="co">#&gt;           Gene Annotation</span>
<span class="co">#&gt; 1 BolC1t00001H  IPR014710</span>
<span class="co">#&gt; 2 BolC1t00001H  IPR018490</span>
<span class="co">#&gt; 3 BolC1t00002H  IPR013057</span>
<span class="co">#&gt; 4 BolC1t00003H  IPR013057</span>
<span class="co">#&gt; 5 BolC1t00004H  IPR005178</span>
<span class="co">#&gt; 6 BolC1t00004H  IPR005178</span></code></pre></div>
<p>If you infer orthogroups with OrthoFinder, you can read and parse the output file <strong>Orthogroups.tsv</strong> with the function <code><a href="../reference/read_orthogroups.html">read_orthogroups()</a></code>. For example:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Path to the Orthogroups.tsv file created by OrthoFinder</span>
<span class="va">og_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Orthogroups.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"cogeqc"</span><span class="op">)</span> 

<span class="co"># Read and parse file</span>
<span class="va">orthogroups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_orthogroups.html">read_orthogroups</a></span><span class="op">(</span><span class="va">og_file</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">orthogroups</span><span class="op">)</span>
<span class="co">#&gt;     Orthogroup Species      Gene</span>
<span class="co">#&gt; 1 HOM05D000001     Ath AT1G02310</span>
<span class="co">#&gt; 2 HOM05D000001     Ath AT1G03510</span>
<span class="co">#&gt; 3 HOM05D000001     Ath AT1G03540</span>
<span class="co">#&gt; 4 HOM05D000001     Ath AT1G04020</span>
<span class="co">#&gt; 5 HOM05D000001     Ath AT1G04840</span>
<span class="co">#&gt; 6 HOM05D000001     Ath AT1G05750</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="assessing-orthogroups">Assessing orthogroups<a class="anchor" aria-label="anchor" href="#assessing-orthogroups"></a>
</h2>
<p>In <code>cogeqc</code>, you can assess orthogroup inference with either a protein domain-based approach or a reference-based approach. Both approaches are described below.</p>
<div class="section level3">
<h3 id="protein-domain-based-orthogroup-assessment">Protein domain-based orthogroup assessment<a class="anchor" aria-label="anchor" href="#protein-domain-based-orthogroup-assessment"></a>
</h3>
<p>To assess the orthogroup inference, you can use the function <code><a href="../reference/assess_orthogroups.html">assess_orthogroups()</a></code>. This function takes as input a list of annotation data frames<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> and an orthogroups data frame, and returns the relative homogeneity frequency<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> of each orthogroup for each species.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create a list of annotation data frames</span>
<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Ath <span class="op">=</span> <span class="va">interpro_ath</span>, Bol <span class="op">=</span> <span class="va">interpro_bol</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="co"># This is what the list must look like</span>
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ Ath:'data.frame': 131661 obs. of  2 variables:</span>
<span class="co">#&gt;   ..$ Gene      : chr [1:131661] "AT1G01010" "AT1G01010" "AT1G01010" "AT1G01020" ...</span>
<span class="co">#&gt;   ..$ Annotation: chr [1:131661] "IPR036093" "IPR003441" "IPR036093" "IPR007290" ...</span>
<span class="co">#&gt;  $ Bol:'data.frame': 212665 obs. of  2 variables:</span>
<span class="co">#&gt;   ..$ Gene      : chr [1:212665] "BolC1t00001H" "BolC1t00001H" "BolC1t00002H" "BolC1t00003H" ...</span>
<span class="co">#&gt;   ..$ Annotation: chr [1:212665] "IPR014710" "IPR018490" "IPR013057" "IPR013057" ...</span>

<span class="va">og_assessment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assess_orthogroups.html">assess_orthogroups</a></span><span class="op">(</span><span class="va">og</span>, <span class="va">annotation</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">og_assessment</span><span class="op">)</span>
<span class="co">#&gt;    Orthogroups      Ath_H      Bol_H     Mean_H</span>
<span class="co">#&gt; 1 HOM05D000001 0.03225806 0.02000000 0.02612903</span>
<span class="co">#&gt; 2 HOM05D000002 0.06666667 0.07142857 0.06904762</span>
<span class="co">#&gt; 3 HOM05D000003 0.10000000 0.07692308 0.08846154</span>
<span class="co">#&gt; 4 HOM05D000004 0.33333333 0.06250000 0.19791667</span>
<span class="co">#&gt; 5 HOM05D000005 0.04761905 0.03333333 0.04047619</span>
<span class="co">#&gt; 6 HOM05D000006 0.14285714 0.12500000 0.13392857</span></code></pre></div>
<p>Now, we can calculate the mean homogeneity for this orthogroup inference.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">og_assessment</span><span class="op">$</span><span class="va">Mean_H</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.5191506</span></code></pre></div>
<p>Ideally, to have a reliable orthogroup inference, you should be able to run OrthoFinder with multiple combinations of parameters and assess each inference with <code><a href="../reference/assess_orthogroups.html">assess_orthogroups()</a></code>. The inference with the highest mean homonegeneity will be the best.<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a></p>
</div>
<div class="section level3">
<h3 id="reference-based-orthogroup-assessment">Reference-based orthogroup assessment<a class="anchor" aria-label="anchor" href="#reference-based-orthogroup-assessment"></a>
</h3>
<p>In some cases, you may want to compare your orthogroup inference to a reference orthogroup inference. To do that, you can use the function <code><a href="../reference/compare_orthogroups.html">compare_orthogroups()</a></code>. For example, let’s simulate a different orthogroup inference by shuffling some rows of the <code>og</code> data frame and comparing it to the original data frame.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>

<span class="co"># Subset the top 5000 rows for demonstration purposes</span>
<span class="va">og_subset</span> <span class="op">&lt;-</span> <span class="va">og</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5000</span>, <span class="op">]</span>
<span class="va">ref</span> <span class="op">&lt;-</span> <span class="va">og_subset</span>

<span class="co"># Shuffle 100 genes to simulate a test set</span>
<span class="va">idx_shuffle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">og_subset</span><span class="op">)</span><span class="op">)</span>, <span class="fl">100</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">test</span> <span class="op">&lt;-</span> <span class="va">og_subset</span>
<span class="va">test</span><span class="op">$</span><span class="va">Gene</span><span class="op">[</span><span class="va">idx_shuffle</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>
  <span class="va">test</span><span class="op">$</span><span class="va">Gene</span><span class="op">[</span><span class="va">idx_shuffle</span><span class="op">]</span>, size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">idx_shuffle</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>

<span class="co"># Compare test set to reference set</span>
<span class="va">comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_orthogroups.html">compare_orthogroups</a></span><span class="op">(</span><span class="va">ref</span>, <span class="va">test</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">)</span>
<span class="co">#&gt;     Orthogroup Preserved</span>
<span class="co">#&gt; 1 HOM05D000001     FALSE</span>
<span class="co">#&gt; 2 HOM05D000002     FALSE</span>
<span class="co">#&gt; 3 HOM05D000003     FALSE</span>
<span class="co">#&gt; 4 HOM05D000004      TRUE</span>
<span class="co">#&gt; 5 HOM05D000005     FALSE</span>
<span class="co">#&gt; 6 HOM05D000006      TRUE</span>

<span class="co"># Calculating percentage of preservation</span>
<span class="va">preserved</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">$</span><span class="va">Preserved</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">$</span><span class="va">Preserved</span><span class="op">)</span>
<span class="va">preserved</span>
<span class="co">#&gt; [1] 0.2702703</span></code></pre></div>
<p>As we can see, 27.03% of the orthogroups in the reference data set are preserved in the shuffled data set.</p>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-summary-statistics">Visualizing summary statistics<a class="anchor" aria-label="anchor" href="#visualizing-summary-statistics"></a>
</h2>
<p>Now that you have identified the best combination of parameters for your orthogroup inference, you can visually explore some of its summary statistics. OrthoFinder automatically saves summary statistics in a directory named <strong>Comparative_Genomics_Statistics</strong>. You can parse this directory in a list of summary statistics with the function <code><a href="../reference/read_orthofinder_stats.html">read_orthofinder_stats()</a></code>. To demonstrate it, let’s read the output of OrthoFinder’s example with model species.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">stats_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"cogeqc"</span><span class="op">)</span>
<span class="va">ortho_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_orthofinder_stats.html">read_orthofinder_stats</a></span><span class="op">(</span><span class="va">stats_dir</span><span class="op">)</span>
<span class="va">ortho_stats</span>
<span class="co">#&gt; $stats</span>
<span class="co">#&gt;                   Species N_genes N_genes_in_OGs Perc_genes_in_OGs N_ssOGs</span>
<span class="co">#&gt; 1             Danio_rerio   30313          28236              93.1     569</span>
<span class="co">#&gt; 2 Drosophila_melanogaster   13931          10674              76.6     675</span>
<span class="co">#&gt; 3            Homo_sapiens   23480          22669              96.5     268</span>
<span class="co">#&gt; 4            Mus_musculus   22859          22006              96.3     243</span>
<span class="co">#&gt; 5       Takifugu_rubripes   20545          19403              94.4     135</span>
<span class="co">#&gt; 6      Xenopus_tropicalis   19987          18755              93.8     234</span>
<span class="co">#&gt;   N_genes_in_ssOGs Perc_genes_in_ssOGs Dups</span>
<span class="co">#&gt; 1             3216                10.6 3353</span>
<span class="co">#&gt; 2             3313                23.8 4527</span>
<span class="co">#&gt; 3             1625                 6.9 2283</span>
<span class="co">#&gt; 4             2022                 8.8 4131</span>
<span class="co">#&gt; 5              446                 2.2 9585</span>
<span class="co">#&gt; 6             1580                 7.9 3650</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $og_overlap</span>
<span class="co">#&gt;                         Danio_rerio Drosophila_melanogaster Homo_sapiens</span>
<span class="co">#&gt; Danio_rerio                   13472                    5872        11365</span>
<span class="co">#&gt; Drosophila_melanogaster        5872                    6651         5866</span>
<span class="co">#&gt; Homo_sapiens                  11365                    5866        14468</span>
<span class="co">#&gt; Mus_musculus                  11345                    5863        14076</span>
<span class="co">#&gt; Takifugu_rubripes             12100                    5810        10994</span>
<span class="co">#&gt; Xenopus_tropicalis            11086                    5725        11478</span>
<span class="co">#&gt;                         Mus_musculus Takifugu_rubripes Xenopus_tropicalis</span>
<span class="co">#&gt; Danio_rerio                    11345             12100              11086</span>
<span class="co">#&gt; Drosophila_melanogaster         5863              5810               5725</span>
<span class="co">#&gt; Homo_sapiens                   14076             10994              11478</span>
<span class="co">#&gt; Mus_musculus                   14411             10976              11446</span>
<span class="co">#&gt; Takifugu_rubripes              10976             12649              10776</span>
<span class="co">#&gt; Xenopus_tropicalis             11446             10776              12302</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $duplications</span>
<span class="co">#&gt;                       Node Duplications_50</span>
<span class="co">#&gt; 1  Drosophila_melanogaster            3353</span>
<span class="co">#&gt; 2             Homo_sapiens            4527</span>
<span class="co">#&gt; 3                       N0              73</span>
<span class="co">#&gt; 4        Takifugu_rubripes            2283</span>
<span class="co">#&gt; 5             Mus_musculus            4131</span>
<span class="co">#&gt; 6              Danio_rerio            9585</span>
<span class="co">#&gt; 7                       N1            2458</span>
<span class="co">#&gt; 8                       N2            1530</span>
<span class="co">#&gt; 9                       N3             195</span>
<span class="co">#&gt; 10                      N4             745</span>
<span class="co">#&gt; 11      Xenopus_tropicalis            3650</span></code></pre></div>
<p>Now, we can use this list to visually explore summary statistics.</p>
<div class="section level3">
<h3 id="species-tree">Species tree<a class="anchor" aria-label="anchor" href="#species-tree"></a>
</h3>
<p>To start, one would usually want to look at the species tree to detect possible issues that would compromise the accuracy of orthologs detection. The tree file can be easily read with <code><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">treeio::read.tree()</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span>
<span class="fu"><a href="../reference/plot_species_tree.html">plot_species_tree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_species_tree-1.png" width="729.166666666667"></p>
<p>You can also include the number of gene duplications in each node.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_species_tree.html">plot_species_tree</a></span><span class="op">(</span><span class="va">tree</span>, stats_list <span class="op">=</span> <span class="va">ortho_stats</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_species_tree_with_dups-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="species-specific-duplications">Species-specific duplications<a class="anchor" aria-label="anchor" href="#species-specific-duplications"></a>
</h3>
<p>The species tree above shows duplications per node, but it does not show species-duplications. To visualize that, you can use the function <code><a href="../reference/plot_duplications.html">plot_duplications()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_duplications.html">plot_duplications</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/unnamed-chunk-2-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="genes-in-orthogroups">Genes in orthogroups<a class="anchor" aria-label="anchor" href="#genes-in-orthogroups"></a>
</h3>
<p>Visualizing the percentage of genes in orthogroups is particularly useful for quality check, since one would usually expect a large percentage of genes in orthogroups, unless there is a very distant species in OrthoFinder’s input proteome data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_genes_in_ogs.html">plot_genes_in_ogs</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_genes_in_ogs-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="species-specific-orthogroups">Species-specific orthogroups<a class="anchor" aria-label="anchor" href="#species-specific-orthogroups"></a>
</h3>
<p>To visualize the number of species-specific orthogroups, use the function <code><a href="../reference/plot_species_specific_ogs.html">plot_species_specific_ogs()</a></code>. This plot can reveal a unique gene repertoire of a particular species if it has a large number of species-specific OGs as compared to the other ones.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_species_specific_ogs.html">plot_species_specific_ogs</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/ssOGs-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="all-in-one">All in one<a class="anchor" aria-label="anchor" href="#all-in-one"></a>
</h3>
<p>To get a complete picture of OrthoFinder results, you can combine all plots together with <code><a href="../reference/plot_orthofinder_stats.html">plot_orthofinder_stats()</a></code>, a wrapper that integrates all previously demonstrated plotting functions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_orthofinder_stats.html">plot_orthofinder_stats</a></span><span class="op">(</span>
  <span class="va">tree</span>, 
  xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.1</span>, <span class="fl">2</span><span class="op">)</span>,
  stats_list <span class="op">=</span> <span class="va">ortho_stats</span>
<span class="op">)</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_orthofinder_stats-1.png" width="1200"></p>
</div>
<div class="section level3">
<h3 id="orthogroup-overlap">Orthogroup overlap<a class="anchor" aria-label="anchor" href="#orthogroup-overlap"></a>
</h3>
<p>You can also visualize a heatmap of pairwise orthogroup overlap across species with <code><a href="../reference/plot_og_overlap.html">plot_og_overlap()</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_og_overlap.html">plot_og_overlap</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_og_overlap-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="orthogroup-size-per-species">Orthogroup size per species<a class="anchor" aria-label="anchor" href="#orthogroup-size-per-species"></a>
</h3>
<p>If you want to take a look at the distribution of OG sizes for each species, you can use the function <code>plot_og_sizes</code>. If you have many extreme values and want to visualize the shape of the distribution in a better way, you can also log transform the OG sizes by setting <code>log = TRUE</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_og_sizes.html">plot_og_sizes</a></span><span class="op">(</span><span class="va">og</span><span class="op">)</span> </code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/og_sizes-1.png" width="729.166666666667"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_og_sizes.html">plot_og_sizes</a></span><span class="op">(</span><span class="va">og</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># natural logarithm scale</span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/og_sizes-2.png" width="729.166666666667"></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.1 (2021-08-10)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] cogeqc_0.99.2    BiocStyle_2.20.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.8.3        ape_5.6-2           lattice_0.20-45    </span>
<span class="co">#&gt;  [4] tidyr_1.2.0         rprojroot_2.0.2     digest_0.6.29      </span>
<span class="co">#&gt;  [7] utf8_1.2.2          plyr_1.8.6          R6_2.5.1           </span>
<span class="co">#&gt; [10] evaluate_0.15       highr_0.9           ggplot2_3.3.5      </span>
<span class="co">#&gt; [13] pillar_1.7.0        ggfun_0.0.5         yulab.utils_0.0.4  </span>
<span class="co">#&gt; [16] rlang_1.0.2         lazyeval_0.2.2      jquerylib_0.1.4    </span>
<span class="co">#&gt; [19] rmarkdown_2.13      pkgdown_2.0.2       labeling_0.4.2     </span>
<span class="co">#&gt; [22] textshaping_0.3.6   desc_1.4.1          stringr_1.4.0      </span>
<span class="co">#&gt; [25] igraph_1.2.11       munsell_0.5.0       compiler_4.1.1     </span>
<span class="co">#&gt; [28] xfun_0.30           pkgconfig_2.0.3     systemfonts_1.0.4  </span>
<span class="co">#&gt; [31] gridGraphics_0.5-1  htmltools_0.5.2     tidyselect_1.1.2   </span>
<span class="co">#&gt; [34] tibble_3.1.6        bookdown_0.25       fansi_1.0.2        </span>
<span class="co">#&gt; [37] crayon_1.5.0        dplyr_1.0.8         grid_4.1.1         </span>
<span class="co">#&gt; [40] nlme_3.1-155        jsonlite_1.8.0      gtable_0.3.0       </span>
<span class="co">#&gt; [43] lifecycle_1.0.1     magrittr_2.0.2      scales_1.1.1       </span>
<span class="co">#&gt; [46] tidytree_0.3.9      cli_3.2.0           stringi_1.7.6      </span>
<span class="co">#&gt; [49] cachem_1.0.6        farver_2.1.0        reshape2_1.4.4     </span>
<span class="co">#&gt; [52] fs_1.5.2            ggtree_3.0.4        bslib_0.3.1        </span>
<span class="co">#&gt; [55] ellipsis_0.3.2      ragg_1.2.2          generics_0.1.2     </span>
<span class="co">#&gt; [58] vctrs_0.3.8         tools_4.1.1         treeio_1.16.2      </span>
<span class="co">#&gt; [61] ggplotify_0.1.0     glue_1.6.2          purrr_0.3.4        </span>
<span class="co">#&gt; [64] parallel_4.1.1      fastmap_1.1.0       yaml_2.3.5         </span>
<span class="co">#&gt; [67] colorspace_2.0-3    BiocManager_1.30.16 aplot_0.1.2        </span>
<span class="co">#&gt; [70] memoise_2.0.1       knitr_1.37          patchwork_1.1.1    </span>
<span class="co">#&gt; [73] sass_0.4.0</span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-emms2019orthofinder" class="csl-entry">
Emms, David M, and Steven Kelly. 2019. <span>“OrthoFinder: Phylogenetic Orthology Inference for Comparative Genomics.”</span> <em>Genome Biology</em> 20 (1): 1–14.
</div>
<div id="ref-li2003orthomcl" class="csl-entry">
Li, Li, Christian J Stoeckert, and David S Roos. 2003. <span>“OrthoMCL: Identification of Ortholog Groups for Eukaryotic Genomes.”</span> <em>Genome Research</em> 13 (9): 2178–89.
</div>
<div id="ref-van2021plaza" class="csl-entry">
Van Bel, Michiel, Francesca Silvestri, Eric M Weitz, Lukasz Kreft, Alexander Botzki, Frederik Coppens, and Klaas Vandepoele. 2021. <span>“PLAZA 5.0: Extending the Scope and Power of Comparative and Functional Genomics in Plants.”</span> <em>Nucleic Acids Research</em>.
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><strong>NOTE:</strong> The names of the list elements must match the species abbreviations in the column <em>Species</em> of the orthogroups data frame. For instance, if your orthogroups data frame contains the species <strong>Ath</strong> and <strong>Bol</strong>, the data frames in the annotation list must be named <strong>Ath</strong> and <strong>Bol</strong> (not necessarily in that order, but with these exact names).<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><strong>Interpretation guide:</strong> Relative homogeneity frequencies range from 0 to 1, with 1 meaning that all genes in that particular orthogroup share the same protein domains, and 0 meaning that each gene in the orthogroup has a different protein domain.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><strong>Friendly tip:</strong> if you want to calculate homogeneity scores using a single species as a proxy (your orthogroups data frame will have only one species), you can use the function <code><a href="../reference/calculate_H.html">calculate_H()</a></code>.<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
