<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cogeqc">
<title>Assessing synteny identification • cogeqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.2/font.css" rel="stylesheet">
<link href="../deps/Source_Code_Pro-0.4.2/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Assessing synteny identification">
<meta property="og:description" content="cogeqc">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-#0092ac"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cogeqc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignette_01_assessing_genome_assembly.html">Assessing genome assembly quality</a>
    <a class="dropdown-item" href="../articles/vignette_02_assessing_orthogroup_inference.html">Assessing orthogroup inference</a>
    <a class="dropdown-item" href="../articles/vignette_03_assessing_synteny.html">Assessing synteny identification</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/almeidasilvaf/cogeqc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Assessing synteny identification</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent,
Belgium<br><h4 data-toc-skip class="author">Yves
Van de Peer</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent,
Belgium<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/cogeqc/blob/HEAD/vignettes/vignette_03_assessing_synteny.Rmd" class="external-link"><code>vignettes/vignette_03_assessing_synteny.Rmd</code></a></small>
      <div class="d-none name"><code>vignette_03_assessing_synteny.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Synteny analysis allows the identification of conserved gene content
and gene order (collinearity) in a genomic segment, and it is often used
to study how genomic rearrangements have shaped genomes during the
course of evolution. However, accurate detection of syntenic blocks is
highly dependent on parameters such as number of top hits in similarity
searches (e.g., with BLAST <span class="citation">(Altschul et al.
1990)</span> or DIAMOND <span class="citation">(Buchfink, Xie, and Huson
2015)</span>), number of anchors, and number of upstream and downstream
genes to search for syntenic blocks. <span class="citation">Zhao and
Schranz (2019)</span> proposed a network-based synteny analysis that
allows the identification of optimal parameters using the network’s
<strong>average clustering coefficient</strong> and <strong>number of
nodes</strong>. The algorithm for synteny network inference has been
implemented in the Bioconductor package <em><a href="https://bioconductor.org/packages/3.15/syntenet" class="external-link">syntenet</a></em>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"cogeqc"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package after installation</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/cogeqc" class="external-link">cogeqc</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># for reproducibility</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>Here, we will use a subset of the synteny network inferred in <span class="citation">Zhao and Schranz (2019)</span> that contains the
synteny network for <em>Brassica oleraceae</em>, <em>B. napus</em>, and
<em>B. rapa</em>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load synteny network for </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">synnet</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">synnet</span><span class="op">)</span></span>
<span><span class="co">#&gt;             anchor1        anchor2</span></span>
<span><span class="co">#&gt; 1 bnp_BnaA01g05780D bol_Bo1g011310</span></span>
<span><span class="co">#&gt; 2 bnp_BnaA01g05800D bol_Bo1g011320</span></span>
<span><span class="co">#&gt; 3 bnp_BnaA01g05810D bol_Bo1g011330</span></span>
<span><span class="co">#&gt; 4 bnp_BnaA01g05820D bol_Bo1g011340</span></span>
<span><span class="co">#&gt; 5 bnp_BnaA01g05830D bol_Bo1g011350</span></span>
<span><span class="co">#&gt; 6 bnp_BnaA01g05840D bol_Bo1g011360</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="network-based-assessment-of-synteny-identification">Network-based assessment of synteny identification<a class="anchor" aria-label="anchor" href="#network-based-assessment-of-synteny-identification"></a>
</h2>
<p>To assess synteny detection, we calculate a synteny network score as
follows:</p>
<p><span class="math display">\[
\begin{aligned}
S &amp;= C N, \text{ where:} \\
\\
C &amp;= \text{Clustering coefficient} \\
N &amp;= \text{Number of nodes}
\end{aligned}
\]</span></p>
<p>The network with the highest score is considered the most accurate.
To score a network, you will use the function
<code><a href="../reference/assess_synnet.html">assess_synnet()</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assess_synnet.html">assess_synnet</a></span><span class="op">(</span><span class="va">synnet</span><span class="op">)</span></span>
<span><span class="co">#&gt;         CC Node_number    Score</span></span>
<span><span class="co">#&gt; 1 0.877912      149144 130935.3</span></span></code></pre></div>
<p>Ideally, you should be able to run a synteny detection program (e.g.,
MCScanX <span class="citation">(Wang et al. 2012)</span> or i-ADHoRE
<span class="citation">(Proost et al. 2012)</span>) with multiple
combinations of parameters, infer a synteny network for each run, and
assess each network to pick the best. To demonstrate it, let’s simulate
different networks through resampling and calculate scores for each of
them with the wrapper function <code><a href="../reference/assess_synnet_list.html">assess_synnet_list()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">net1</span> <span class="op">&lt;-</span> <span class="va">synnet</span></span>
<span><span class="va">net2</span> <span class="op">&lt;-</span> <span class="va">synnet</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10000</span>, <span class="fl">500</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">net3</span> <span class="op">&lt;-</span> <span class="va">synnet</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10000</span>, <span class="fl">1000</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">synnet_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  net1 <span class="op">=</span> <span class="va">net1</span>, </span>
<span>  net2 <span class="op">=</span> <span class="va">net2</span>, </span>
<span>  net3 <span class="op">=</span> <span class="va">net3</span></span>
<span><span class="op">)</span></span>
<span><span class="va">synnet_assesment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assess_synnet_list.html">assess_synnet_list</a></span><span class="op">(</span><span class="va">synnet_list</span><span class="op">)</span></span>
<span><span class="va">synnet_assesment</span></span>
<span><span class="co">#&gt;          CC Node_number    Score Network</span></span>
<span><span class="co">#&gt; 1 0.8779120      149144 130935.3    net1</span></span>
<span><span class="co">#&gt; 2 0.8769428      149133 130781.1    net2</span></span>
<span><span class="co">#&gt; 3 0.8758974      149114 130608.6    net3</span></span>
<span></span>
<span><span class="co"># Determine the best network</span></span>
<span><span class="va">synnet_assesment</span><span class="op">$</span><span class="va">Network</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">synnet_assesment</span><span class="op">$</span><span class="va">Score</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] "net1"</span></span></code></pre></div>
<p>As you can see, the first (original) network is the best one, as it
has the highest score.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] cogeqc_1.1.6     BiocStyle_2.24.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] Rcpp_1.0.9             ape_5.6-2              lattice_0.20-45       </span></span>
<span><span class="co">#&gt;  [4] tidyr_1.2.1            Biostrings_2.64.1      rprojroot_2.0.3       </span></span>
<span><span class="co">#&gt;  [7] digest_0.6.29          utf8_1.2.2             plyr_1.8.7            </span></span>
<span><span class="co">#&gt; [10] R6_2.5.1               GenomeInfoDb_1.32.4    stats4_4.2.1          </span></span>
<span><span class="co">#&gt; [13] evaluate_0.16          ggplot2_3.3.6          pillar_1.8.1          </span></span>
<span><span class="co">#&gt; [16] ggfun_0.0.7            yulab.utils_0.0.5      zlibbioc_1.42.0       </span></span>
<span><span class="co">#&gt; [19] rlang_1.0.5            lazyeval_0.2.2         jquerylib_0.1.4       </span></span>
<span><span class="co">#&gt; [22] S4Vectors_0.34.0       rmarkdown_2.16         pkgdown_2.0.6         </span></span>
<span><span class="co">#&gt; [25] textshaping_0.3.6      desc_1.4.2             stringr_1.4.1         </span></span>
<span><span class="co">#&gt; [28] igraph_1.3.5           RCurl_1.98-1.8         munsell_0.5.0         </span></span>
<span><span class="co">#&gt; [31] compiler_4.2.1         xfun_0.33              gridGraphics_0.5-1    </span></span>
<span><span class="co">#&gt; [34] pkgconfig_2.0.3        systemfonts_1.0.4      BiocGenerics_0.42.0   </span></span>
<span><span class="co">#&gt; [37] htmltools_0.5.3        tidyselect_1.1.2       tibble_3.1.8          </span></span>
<span><span class="co">#&gt; [40] GenomeInfoDbData_1.2.8 bookdown_0.29          IRanges_2.30.1        </span></span>
<span><span class="co">#&gt; [43] fansi_1.0.3            crayon_1.5.1           dplyr_1.0.10          </span></span>
<span><span class="co">#&gt; [46] bitops_1.0-7           grid_4.2.1             nlme_3.1-159          </span></span>
<span><span class="co">#&gt; [49] jsonlite_1.8.0         gtable_0.3.1           lifecycle_1.0.2       </span></span>
<span><span class="co">#&gt; [52] magrittr_2.0.3         scales_1.2.1           tidytree_0.4.0        </span></span>
<span><span class="co">#&gt; [55] cli_3.4.0              stringi_1.7.8          cachem_1.0.6          </span></span>
<span><span class="co">#&gt; [58] reshape2_1.4.4         XVector_0.36.0         fs_1.5.2              </span></span>
<span><span class="co">#&gt; [61] ggtree_3.4.2           bslib_0.4.0            ragg_1.2.2            </span></span>
<span><span class="co">#&gt; [64] vctrs_0.4.1            generics_0.1.3         treeio_1.20.2         </span></span>
<span><span class="co">#&gt; [67] tools_4.2.1            ggplotify_0.1.0        glue_1.6.2            </span></span>
<span><span class="co">#&gt; [70] purrr_0.3.4            parallel_4.2.1         fastmap_1.1.0         </span></span>
<span><span class="co">#&gt; [73] yaml_2.3.5             colorspace_2.0-3       BiocManager_1.30.18   </span></span>
<span><span class="co">#&gt; [76] aplot_0.1.7            memoise_2.0.1          knitr_1.40            </span></span>
<span><span class="co">#&gt; [79] patchwork_1.1.2        sass_0.4.2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-altschul1990basic" class="csl-entry">
Altschul, Stephen F, Warren Gish, Webb Miller, Eugene W Myers, and David
J Lipman. 1990. <span>“Basic Local Alignment Search Tool.”</span>
<em>Journal of Molecular Biology</em> 215 (3): 403–10.
</div>
<div id="ref-buchfink2015fast" class="csl-entry">
Buchfink, Benjamin, Chao Xie, and Daniel H Huson. 2015. <span>“Fast and
Sensitive Protein Alignment Using DIAMOND.”</span> <em>Nature
Methods</em> 12 (1): 59–60.
</div>
<div id="ref-proost2012adhore" class="csl-entry">
Proost, Sebastian, Jan Fostier, Dieter De Witte, Bart Dhoedt, Piet
Demeester, Yves Van de Peer, and Klaas Vandepoele. 2012. <span>“I-ADHoRe
3.0—Fast and Sensitive Detection of Genomic Homology in Extremely Large
Data Sets.”</span> <em>Nucleic Acids Research</em> 40 (2): e11–11.
</div>
<div id="ref-wang2012mcscanx" class="csl-entry">
Wang, Yupeng, Haibao Tang, Jeremy D DeBarry, Xu Tan, Jingping Li, Xiyin
Wang, Tae-ho Lee, et al. 2012. <span>“MCScanX: A Toolkit for Detection
and Evolutionary Analysis of Gene Synteny and Collinearity.”</span>
<em>Nucleic Acids Research</em> 40 (7): e49–49.
</div>
<div id="ref-zhao2019network" class="csl-entry">
Zhao, Tao, and M Eric Schranz. 2019. <span>“Network-Based Microsynteny
Analysis Identifies Major Differences and Genomic Outliers in Mammalian
and Angiosperm Genomes.”</span> <em>Proceedings of the National Academy
of Sciences</em> 116 (6): 2165–74.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
