<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Assessing synteny identification • cogeqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Assessing synteny identification">
<meta property="og:description" content="cogeqc">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cogeqc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/vignette_00_note_on_reproducibility.html">A note on reproducibility</a>
    </li>
    <li>
      <a href="../articles/vignette_01_assessing_genome_assembly.html">Assessing genome assembly quality</a>
    </li>
    <li>
      <a href="../articles/vignette_02_assessing_orthogroup_inference.html">Assessing orthogroup inference</a>
    </li>
    <li>
      <a href="../articles/vignette_03_assessing_synteny.html">Assessing synteny identification</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/almeidasilvaf/cogeqc/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Assessing synteny identification</h1>
                        <h4 data-toc-skip class="author">Fabricio Almeida-Silva</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent, Belgium<br><h4 data-toc-skip class="author">Yves Van de Peer</h4>
            <address class="author_afil">
      Department of Plant Biotechnology and Bioinformatics, Ghent University, Ghent, BelgiumVIB-UGent Center for Plant Systems Biology, Ghent, BelgiumCollege of Horticulture, Academy for Advanced Interdisciplinary Studies, Nanjing Agricultural University, Nanjing, ChinaCenter for Microbial Ecology and Genomics, Department of Biochemistry, Genetics and Microbiology, University of Pretoria, Pretoria, South Africa<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/cogeqc/blob/HEAD/vignettes/vignette_03_assessing_synteny.Rmd" class="external-link"><code>vignettes/vignette_03_assessing_synteny.Rmd</code></a></small>
      <div class="hidden name"><code>vignette_03_assessing_synteny.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Synteny analysis allows the identification of conserved gene content and gene order (collinearity) in a genomic segment, and it is often used to study how genomic rearrangements have shaped chromosomes during the course of evolution. However, accurate detection of syntenic blocks is highly dependent on parameters such as number of top hits in similarity searches (e.g., with BLAST <span class="citation">(Altschul et al. 1990)</span> or DIAMOND <span class="citation">(Buchfink, Xie, and Huson 2015)</span>), number of anchors, and number of upstream and downstream genes to search for syntenic blocks. <span class="citation">Zhao and Schranz (2019)</span> proposed a network-based synteny analysis that allows the identification of optimal parameters using the network’s <strong>average clustering coefficient</strong> and <strong>number of nodes</strong>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"cogeqc"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load package after installation</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/cogeqc" class="external-link">cogeqc</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> <span class="co"># for reproducibility</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>Here, we will use a subset of the synteny network inferred in <span class="citation">Zhao and Schranz (2019)</span> that contains the synteny network for <em>Brassica oleraceae</em>, <em>B. napus</em>, and <em>B. rapa</em>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load synteny network for </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">synnet</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">synnet</span><span class="op">)</span>
<span class="co">#&gt;             anchor1        anchor2</span>
<span class="co">#&gt; 1 bnp_BnaA01g05780D bol_Bo1g011310</span>
<span class="co">#&gt; 2 bnp_BnaA01g05800D bol_Bo1g011320</span>
<span class="co">#&gt; 3 bnp_BnaA01g05810D bol_Bo1g011330</span>
<span class="co">#&gt; 4 bnp_BnaA01g05820D bol_Bo1g011340</span>
<span class="co">#&gt; 5 bnp_BnaA01g05830D bol_Bo1g011350</span>
<span class="co">#&gt; 6 bnp_BnaA01g05840D bol_Bo1g011360</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="network-based-assessment-of-synteny-identification">Network-based assessment of synteny identification<a class="anchor" aria-label="anchor" href="#network-based-assessment-of-synteny-identification"></a>
</h2>
<p>To assess synteny detection, we calculate a synteny network score as follows:</p>
<p><span class="math display">\[
\begin{aligned}
S &amp;= C N, \text{ where:} \\
\\
C &amp;= \text{Clustering coefficient} \\
N &amp;= \text{Number of nodes}
\end{aligned}
\]</span></p>
<p>The network with the highest score is considered the most accurate. To score a network, you will use the function <code><a href="../reference/assess_synnet.html">assess_synnet()</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/assess_synnet.html">assess_synnet</a></span><span class="op">(</span><span class="va">synnet</span><span class="op">)</span>
<span class="co">#&gt;         CC Node_number    Score</span>
<span class="co">#&gt; 1 0.877912      149144 130935.3</span></code></pre></div>
<p>Ideally, you should be able to run a synteny detection program (e.g., MCScanX <span class="citation">(Wang et al. 2012)</span> or i-ADHoRE <span class="citation">(Proost et al. 2012)</span>) with multiple combinations of parameters, infer a synteny network for each run, and assess each network to pick the best. To demonstrate it, let’s simulate different networks through resampling and calculate scores for each of them with the wrapper function <code><a href="../reference/assess_synnet_list.html">assess_synnet_list()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">net1</span> <span class="op">&lt;-</span> <span class="va">synnet</span>
<span class="va">net2</span> <span class="op">&lt;-</span> <span class="va">synnet</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10000</span>, <span class="fl">500</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">net3</span> <span class="op">&lt;-</span> <span class="va">synnet</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10000</span>, <span class="fl">1000</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">synnet_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  net1 <span class="op">=</span> <span class="va">net1</span>, 
  net2 <span class="op">=</span> <span class="va">net2</span>, 
  net3 <span class="op">=</span> <span class="va">net3</span>
<span class="op">)</span>
<span class="va">synnet_assesment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assess_synnet_list.html">assess_synnet_list</a></span><span class="op">(</span><span class="va">synnet_list</span><span class="op">)</span>
<span class="va">synnet_assesment</span>
<span class="co">#&gt;          CC Node_number    Score Network</span>
<span class="co">#&gt; 1 0.8779120      149144 130935.3    net1</span>
<span class="co">#&gt; 2 0.8769428      149133 130781.1    net2</span>
<span class="co">#&gt; 3 0.8758974      149114 130608.6    net3</span>

<span class="co"># Determine the best network</span>
<span class="va">synnet_assesment</span><span class="op">$</span><span class="va">Network</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.min.html" class="external-link">which.max</a></span><span class="op">(</span><span class="va">synnet_assesment</span><span class="op">$</span><span class="va">Score</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; [1] "net1"</span></code></pre></div>
<p>As you can see, the first (original) network is the best one, as it has the highest score.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.1 (2021-08-10)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] cogeqc_0.99.0    BiocStyle_2.20.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Rcpp_1.0.8          ape_5.6-2           lattice_0.20-45    </span>
<span class="co">#&gt;  [4] tidyr_1.2.0         png_0.1-7           rprojroot_2.0.2    </span>
<span class="co">#&gt;  [7] digest_0.6.29       utf8_1.2.2          plyr_1.8.6         </span>
<span class="co">#&gt; [10] R6_2.5.1            Herper_1.2.0        evaluate_0.15      </span>
<span class="co">#&gt; [13] ggplot2_3.3.5       pillar_1.7.0        ggfun_0.0.5        </span>
<span class="co">#&gt; [16] yulab.utils_0.0.4   rlang_1.0.2         lazyeval_0.2.2     </span>
<span class="co">#&gt; [19] jquerylib_0.1.4     Matrix_1.4-0        reticulate_1.24    </span>
<span class="co">#&gt; [22] rmarkdown_2.13      pkgdown_2.0.2       textshaping_0.3.6  </span>
<span class="co">#&gt; [25] desc_1.4.1          stringr_1.4.0       igraph_1.2.11      </span>
<span class="co">#&gt; [28] munsell_0.5.0       compiler_4.1.1      xfun_0.30          </span>
<span class="co">#&gt; [31] pkgconfig_2.0.3     systemfonts_1.0.4   gridGraphics_0.5-1 </span>
<span class="co">#&gt; [34] htmltools_0.5.2     tidyselect_1.1.2    tibble_3.1.6       </span>
<span class="co">#&gt; [37] bookdown_0.24       fansi_1.0.2         crayon_1.5.0       </span>
<span class="co">#&gt; [40] dplyr_1.0.8         withr_2.5.0         grid_4.1.1         </span>
<span class="co">#&gt; [43] nlme_3.1-155        jsonlite_1.8.0      gtable_0.3.0       </span>
<span class="co">#&gt; [46] lifecycle_1.0.1     magrittr_2.0.2      scales_1.1.1       </span>
<span class="co">#&gt; [49] tidytree_0.3.9      cli_3.2.0           stringi_1.7.6      </span>
<span class="co">#&gt; [52] cachem_1.0.6        reshape2_1.4.4      fs_1.5.2           </span>
<span class="co">#&gt; [55] ggtree_3.0.4        bslib_0.3.1         ellipsis_0.3.2     </span>
<span class="co">#&gt; [58] ragg_1.2.2          generics_0.1.2      vctrs_0.3.8        </span>
<span class="co">#&gt; [61] rjson_0.2.21        treeio_1.16.2       tools_4.1.1        </span>
<span class="co">#&gt; [64] ggplotify_0.1.0     glue_1.6.2          purrr_0.3.4        </span>
<span class="co">#&gt; [67] parallel_4.1.1      fastmap_1.1.0       yaml_2.3.5         </span>
<span class="co">#&gt; [70] colorspace_2.0-3    BiocManager_1.30.16 aplot_0.1.2        </span>
<span class="co">#&gt; [73] memoise_2.0.1       knitr_1.37          patchwork_1.1.1    </span>
<span class="co">#&gt; [76] sass_0.4.0</span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-altschul1990basic" class="csl-entry">
Altschul, Stephen F, Warren Gish, Webb Miller, Eugene W Myers, and David J Lipman. 1990. <span>“Basic Local Alignment Search Tool.”</span> <em>Journal of Molecular Biology</em> 215 (3): 403–10.
</div>
<div id="ref-buchfink2015fast" class="csl-entry">
Buchfink, Benjamin, Chao Xie, and Daniel H Huson. 2015. <span>“Fast and Sensitive Protein Alignment Using DIAMOND.”</span> <em>Nature Methods</em> 12 (1): 59–60.
</div>
<div id="ref-proost2012adhore" class="csl-entry">
Proost, Sebastian, Jan Fostier, Dieter De Witte, Bart Dhoedt, Piet Demeester, Yves Van de Peer, and Klaas Vandepoele. 2012. <span>“I-ADHoRe 3.0—Fast and Sensitive Detection of Genomic Homology in Extremely Large Data Sets.”</span> <em>Nucleic Acids Research</em> 40 (2): e11–11.
</div>
<div id="ref-wang2012mcscanx" class="csl-entry">
Wang, Yupeng, Haibao Tang, Jeremy D DeBarry, Xu Tan, Jingping Li, Xiyin Wang, Tae-ho Lee, et al. 2012. <span>“MCScanX: A Toolkit for Detection and Evolutionary Analysis of Gene Synteny and Collinearity.”</span> <em>Nucleic Acids Research</em> 40 (7): e49–49.
</div>
<div id="ref-zhao2019network" class="csl-entry">
Zhao, Tao, and M Eric Schranz. 2019. <span>“Network-Based Microsynteny Analysis Identifies Major Differences and Genomic Outliers in Mammalian and Angiosperm Genomes.”</span> <em>Proceedings of the National Academy of Sciences</em> 116 (6): 2165–74.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
