<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cogeqc">
<title>Assessing orthogroup inference • cogeqc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.5/font.css" rel="stylesheet">
<link href="../deps/Source_Code_Pro-0.4.5/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Assessing orthogroup inference">
<meta property="og:description" content="cogeqc">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-#0092ac"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cogeqc</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/vignette_01_assessing_genome_assembly.html">Assessing genome assembly and annotation quality</a>
    <a class="dropdown-item" href="../articles/vignette_02_assessing_orthogroup_inference.html">Assessing orthogroup inference</a>
    <a class="dropdown-item" href="../articles/vignette_03_assessing_synteny.html">Assessing synteny identification</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/almeidasilvaf/cogeqc/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Assessing orthogroup inference</h1>
                        <h4 data-toc-skip class="author">Fabricio
Almeida-Silva</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent,
Belgium<br><h4 data-toc-skip class="author">Yves
Van de Peer</h4>
            <address class="author_afil">
      VIB-UGent Center for Plant Systems Biology, Ghent University,
Ghent, Belgium<br><small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/cogeqc/blob/HEAD/vignettes/vignette_02_assessing_orthogroup_inference.Rmd" class="external-link"><code>vignettes/vignette_02_assessing_orthogroup_inference.Rmd</code></a></small>
      <div class="d-none name"><code>vignette_02_assessing_orthogroup_inference.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The identification of groups of homologous genes within and across
species is a powerful tool for evolutionary genomics. The most widely
used tools to identify orthogroups (i.e., groups of orthologous genes)
are OrthoFinder <span class="citation">(Emms and Kelly 2019)</span> and
OrthoMCL <span class="citation">(Li, Stoeckert, and Roos 2003)</span>.
However, these tools generate different results depending on the
parameters used, such as <em>mcl inflation parameter</em>,
<em>E-value</em>, <em>maximum number of hits</em>, and others. Here, we
propose a protein domain-aware assessment of orthogroup inference. The
goal is to maximize the percentage of shared protein domains for genes
in the same orthogroup.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">'BiocManager'</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">'BiocManager'</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"cogeqc"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package after installation</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/cogeqc" class="external-link">cogeqc</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>Here, we will use orthogroups from the PLAZA 5.0 database <span class="citation">(Van Bel et al. 2021)</span>, inferred with OrthoFinder
<span class="citation">(Emms and Kelly 2019)</span>. For the purpose of
demonstration, the complete dataset was filtered to only keep
orthogroups for the Brassicaceae species <em>Arabidopsis thaliana</em>
and <em>Brassica oleraceae</em>. Interpro domain annotations were also
retrieved from PLAZA 5.0.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Orthogroups for Arabidopsis thaliana and Brassica oleraceae</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">og</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">og</span><span class="op">)</span></span>
<span><span class="co">#&gt;     Orthogroup Species      Gene</span></span>
<span><span class="co">#&gt; 1 HOM05D000001     Ath AT1G02310</span></span>
<span><span class="co">#&gt; 2 HOM05D000001     Ath AT1G03510</span></span>
<span><span class="co">#&gt; 3 HOM05D000001     Ath AT1G03540</span></span>
<span><span class="co">#&gt; 4 HOM05D000001     Ath AT1G04020</span></span>
<span><span class="co">#&gt; 5 HOM05D000001     Ath AT1G04840</span></span>
<span><span class="co">#&gt; 6 HOM05D000001     Ath AT1G05750</span></span>
<span></span>
<span><span class="co"># Interpro domain annotations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">interpro_ath</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">interpro_bol</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">interpro_ath</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Gene Annotation</span></span>
<span><span class="co">#&gt; 1 AT1G01010  IPR036093</span></span>
<span><span class="co">#&gt; 2 AT1G01010  IPR003441</span></span>
<span><span class="co">#&gt; 3 AT1G01010  IPR036093</span></span>
<span><span class="co">#&gt; 4 AT1G01020  IPR007290</span></span>
<span><span class="co">#&gt; 5 AT1G01020  IPR007290</span></span>
<span><span class="co">#&gt; 6 AT1G01030  IPR003340</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">interpro_bol</span><span class="op">)</span></span>
<span><span class="co">#&gt;           Gene Annotation</span></span>
<span><span class="co">#&gt; 1 BolC1t00001H  IPR014710</span></span>
<span><span class="co">#&gt; 2 BolC1t00001H  IPR018490</span></span>
<span><span class="co">#&gt; 3 BolC1t00002H  IPR013057</span></span>
<span><span class="co">#&gt; 4 BolC1t00003H  IPR013057</span></span>
<span><span class="co">#&gt; 5 BolC1t00004H  IPR005178</span></span>
<span><span class="co">#&gt; 6 BolC1t00004H  IPR005178</span></span></code></pre></div>
<p>If you infer orthogroups with OrthoFinder, you can read and parse the
output file <strong>Orthogroups.tsv</strong> with the function
<code><a href="../reference/read_orthogroups.html">read_orthogroups()</a></code>. For example:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Path to the Orthogroups.tsv file created by OrthoFinder</span></span>
<span><span class="va">og_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Orthogroups.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"cogeqc"</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Read and parse file</span></span>
<span><span class="va">orthogroups</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_orthogroups.html">read_orthogroups</a></span><span class="op">(</span><span class="va">og_file</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">orthogroups</span><span class="op">)</span></span>
<span><span class="co">#&gt;     Orthogroup Species      Gene</span></span>
<span><span class="co">#&gt; 1 HOM05D000001     Ath AT1G02310</span></span>
<span><span class="co">#&gt; 2 HOM05D000001     Ath AT1G03510</span></span>
<span><span class="co">#&gt; 3 HOM05D000001     Ath AT1G03540</span></span>
<span><span class="co">#&gt; 4 HOM05D000001     Ath AT1G04020</span></span>
<span><span class="co">#&gt; 5 HOM05D000001     Ath AT1G04840</span></span>
<span><span class="co">#&gt; 6 HOM05D000001     Ath AT1G05750</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="assessing-orthogroups">Assessing orthogroups<a class="anchor" aria-label="anchor" href="#assessing-orthogroups"></a>
</h2>
<p>In <code>cogeqc</code>, you can assess orthogroup inference with
either a protein domain-based approach or a reference-based approach.
Both approaches are described below.</p>
<div class="section level3">
<h3 id="protein-domain-based-orthogroup-assessment">Protein domain-based orthogroup assessment<a class="anchor" aria-label="anchor" href="#protein-domain-based-orthogroup-assessment"></a>
</h3>
<p>The protein domain-based assessment of orthogroups is based on the
formula below:</p>
<p><span class="math display">\[
\begin{aligned}
Scores &amp;= \frac{Homogeneity}{Dispersal}
\\
\end{aligned}
\]</span></p>
<p>The numerator, <span class="math inline">\(homogeneity\)</span>, is
the mean Sorensen-Dice index for all pairwise combinations of genes in
an orthogroup. The Sorensen-Dice index measures how similar two genes
are, and it ranges from 0 to 1, with 0 meaning that a gene pair does not
share any protein domain, and 1 meaning that it shares all protein
domains. In a formal definition:</p>
<p><span class="math display">\[
\begin{aligned}
Homogeneity &amp;=  \frac{1}{N_{pairs}} \sum_{i=1}^{N_{pairs}} SDI_{i}
\\
\\
SDI(A,B) &amp;= \frac{2 \left| A \cap B \right|}{ \left|A \right| +
\left| B \right|}
\end{aligned}
\]</span></p>
<p>where A and B are the set of protein domains associated to genes A
and B. This way, if all genes in an orthogroup have the same protein
domains, it will have <span class="math inline">\(homogeneity =
1\)</span>. If each gene has a different protein domain, the orthogroup
will have <span class="math inline">\(homogeneity = 0\)</span>. If only
some gene pairs share the same domain, <span class="math inline">\(homogeneity\)</span> will be somewhere between 0
and 1.</p>
<p>The denominator, <span class="math inline">\(dispersal\)</span>, aims
to correct for overclustering (i.e., orthogroup assignments that break
“true” gene families into an artificially large number of smaller
subfamilies). It is the mean number of orthogroups containing the same
protein domain corrected by the number of orthogroup. Formally:</p>
<p><span class="math display">\[
\begin{aligned}
Dispersal &amp;=  \frac{1}{N_{domains} N_{OG}}
\sum_{i=1}^{N_{domains}}D_{i} \\
\\
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(N_{OG}\)</span> is the number of
orthogroups, and <span class="math inline">\(D_{i}\)</span> is the
number of orthogroups containing the protein domain <span class="math inline">\(i\)</span>. This term penalizes orthogroup
assignments where the same protein domains appears in multiple
orthogroups. As orthogroups represent groups of genes that evolved from
a common ancestor, a protein domain being present in multiple
orthogroups indicates that this domain evolved multiple times in an
independent way, which is not reasonable from a phylogenetic point of
view, despite convergent evolution.</p>
<p>To calculate scores for each orthogroup, you can use the function
<code><a href="../reference/assess_orthogroups.html">assess_orthogroups()</a></code>. This function takes as input a list
of annotation data frames<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;&lt;strong&gt;NOTE:&lt;/strong&gt; The names of the list elements
must match the species abbreviations in the column &lt;em&gt;Species&lt;/em&gt; of
the orthogroups data frame. For instance, if your orthogroups data frame
contains the species &lt;strong&gt;Ath&lt;/strong&gt; and &lt;strong&gt;Bol&lt;/strong&gt;, the
data frames in the annotation list must be named &lt;strong&gt;Ath&lt;/strong&gt;
and &lt;strong&gt;Bol&lt;/strong&gt; (not necessarily in that order, but with these
exact names).&lt;/p&gt;"><sup>1</sup></a> and an orthogroups data frame, and returns
the relative homogeneity scores of each orthogroup for each species.
Note that if you don’t want to take the dispersal into account, you can
set <code>correct_overclustering = FALSE</code>. This will ignore the
denominator of the score formula.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a list of annotation data frames</span></span>
<span><span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Ath <span class="op">=</span> <span class="va">interpro_ath</span>, Bol <span class="op">=</span> <span class="va">interpro_bol</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="co"># This is what the list must look like</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ Ath:'data.frame': 131661 obs. of  2 variables:</span></span>
<span><span class="co">#&gt;   ..$ Gene      : chr [1:131661] "AT1G01010" "AT1G01010" "AT1G01010" "AT1G01020" ...</span></span>
<span><span class="co">#&gt;   ..$ Annotation: chr [1:131661] "IPR036093" "IPR003441" "IPR036093" "IPR007290" ...</span></span>
<span><span class="co">#&gt;  $ Bol:'data.frame': 212665 obs. of  2 variables:</span></span>
<span><span class="co">#&gt;   ..$ Gene      : chr [1:212665] "BolC1t00001H" "BolC1t00001H" "BolC1t00002H" "BolC1t00003H" ...</span></span>
<span><span class="co">#&gt;   ..$ Annotation: chr [1:212665] "IPR014710" "IPR018490" "IPR013057" "IPR013057" ...</span></span>
<span></span>
<span><span class="va">og_assessment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assess_orthogroups.html">assess_orthogroups</a></span><span class="op">(</span><span class="va">og</span>, <span class="va">annotation</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">og_assessment</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Orthogroups Ath_score Bol_score Mean_score Median_score</span></span>
<span><span class="co">#&gt; 1 HOM05D000001  283.3132  271.9950   277.6541     277.6541</span></span>
<span><span class="co">#&gt; 2 HOM05D000002  129.9598  515.2557   322.6078     322.6078</span></span>
<span><span class="co">#&gt; 3 HOM05D000003  889.1268  848.1947   868.6607     868.6607</span></span>
<span><span class="co">#&gt; 4 HOM05D000004    0.0000  940.5871   470.2935     470.2935</span></span>
<span><span class="co">#&gt; 5 HOM05D000005 1135.8799  808.1998   972.0398     972.0398</span></span>
<span><span class="co">#&gt; 6 HOM05D000006 2820.8337  899.6528  1860.2433    1860.2433</span></span></code></pre></div>
<p>Now, we can calculate the mean score for this orthogroup
inference.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">og_assessment</span><span class="op">$</span><span class="va">Mean_score</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1686.855</span></span></code></pre></div>
<p>Ideally, to have a reliable orthogroup inference, you should be able
to run OrthoFinder with multiple combinations of parameters and assess
each inference with <code><a href="../reference/assess_orthogroups.html">assess_orthogroups()</a></code>. The inference
with the highest mean homonegeneity will be the best.<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;strong&gt;Friendly tip:&lt;/strong&gt; if you want to calculate
homogeneity scores using a single species as a proxy (your orthogroups
data frame will have only one species), you can use the function
&lt;code&gt;&lt;a href="../reference/calculate_H.html"&gt;calculate_H()&lt;/a&gt;&lt;/code&gt;.&lt;/p&gt;'><sup>2</sup></a></p>
</div>
<div class="section level3">
<h3 id="reference-based-orthogroup-assessment">Reference-based orthogroup assessment<a class="anchor" aria-label="anchor" href="#reference-based-orthogroup-assessment"></a>
</h3>
<p>In some cases, you may want to compare your orthogroup inference to a
reference orthogroup inference. To do that, you can use the function
<code><a href="../reference/compare_orthogroups.html">compare_orthogroups()</a></code>. For example, let’s simulate a
different orthogroup inference by shuffling some rows of the
<code>og</code> data frame and compare it to the original data
frame.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Subset the top 5000 rows for demonstration purposes</span></span>
<span><span class="va">og_subset</span> <span class="op">&lt;-</span> <span class="va">og</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5000</span>, <span class="op">]</span></span>
<span><span class="va">ref</span> <span class="op">&lt;-</span> <span class="va">og_subset</span></span>
<span></span>
<span><span class="co"># Shuffle 100 genes to simulate a test set</span></span>
<span><span class="va">idx_shuffle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">og_subset</span><span class="op">)</span><span class="op">)</span>, <span class="fl">100</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">test</span> <span class="op">&lt;-</span> <span class="va">og_subset</span></span>
<span><span class="va">test</span><span class="op">$</span><span class="va">Gene</span><span class="op">[</span><span class="va">idx_shuffle</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span></span>
<span>  <span class="va">test</span><span class="op">$</span><span class="va">Gene</span><span class="op">[</span><span class="va">idx_shuffle</span><span class="op">]</span>, size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">idx_shuffle</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare test set to reference set</span></span>
<span><span class="va">comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compare_orthogroups.html">compare_orthogroups</a></span><span class="op">(</span><span class="va">ref</span>, <span class="va">test</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">)</span></span>
<span><span class="co">#&gt;     Orthogroup Preserved</span></span>
<span><span class="co">#&gt; 1 HOM05D000001     FALSE</span></span>
<span><span class="co">#&gt; 2 HOM05D000002     FALSE</span></span>
<span><span class="co">#&gt; 3 HOM05D000003     FALSE</span></span>
<span><span class="co">#&gt; 4 HOM05D000004      TRUE</span></span>
<span><span class="co">#&gt; 5 HOM05D000005     FALSE</span></span>
<span><span class="co">#&gt; 6 HOM05D000006      TRUE</span></span>
<span></span>
<span><span class="co"># Calculating percentage of preservation</span></span>
<span><span class="va">preserved</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">$</span><span class="va">Preserved</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">$</span><span class="va">Preserved</span><span class="op">)</span></span>
<span><span class="va">preserved</span></span>
<span><span class="co">#&gt; [1] 0.2702703</span></span></code></pre></div>
<p>As we can see, 27.03% of the orthogroups in the reference data set
are preserved in the shuffled data set.</p>
</div>
</div>
<div class="section level2">
<h2 id="visualizing-summary-statistics">Visualizing summary statistics<a class="anchor" aria-label="anchor" href="#visualizing-summary-statistics"></a>
</h2>
<p>Now that you have identified the best combination of parameters for
your orthogroup inference, you can visually explore some of its summary
statistics. OrthoFinder automatically saves summary statistics in a
directory named <strong>Comparative_Genomics_Statistics</strong>. You
can parse this directory in a list of summary statistics with the
function <code><a href="../reference/read_orthofinder_stats.html">read_orthofinder_stats()</a></code>. To demonstrate it, let’s
read the output of OrthoFinder’s example with model species.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"cogeqc"</span><span class="op">)</span></span>
<span><span class="va">ortho_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_orthofinder_stats.html">read_orthofinder_stats</a></span><span class="op">(</span><span class="va">stats_dir</span><span class="op">)</span></span>
<span><span class="va">ortho_stats</span></span>
<span><span class="co">#&gt; $stats</span></span>
<span><span class="co">#&gt;                   Species N_genes N_genes_in_OGs Perc_genes_in_OGs N_ssOGs</span></span>
<span><span class="co">#&gt; 1             Danio_rerio   30313          28236              93.1     569</span></span>
<span><span class="co">#&gt; 2 Drosophila_melanogaster   13931          10674              76.6     675</span></span>
<span><span class="co">#&gt; 3            Homo_sapiens   23480          22669              96.5     268</span></span>
<span><span class="co">#&gt; 4            Mus_musculus   22859          22006              96.3     243</span></span>
<span><span class="co">#&gt; 5       Takifugu_rubripes   20545          19403              94.4     135</span></span>
<span><span class="co">#&gt; 6      Xenopus_tropicalis   19987          18755              93.8     234</span></span>
<span><span class="co">#&gt;   N_genes_in_ssOGs Perc_genes_in_ssOGs Dups</span></span>
<span><span class="co">#&gt; 1             3216                10.6 9585</span></span>
<span><span class="co">#&gt; 2             3313                23.8 3353</span></span>
<span><span class="co">#&gt; 3             1625                 6.9 4527</span></span>
<span><span class="co">#&gt; 4             2022                 8.8 4131</span></span>
<span><span class="co">#&gt; 5              446                 2.2 2283</span></span>
<span><span class="co">#&gt; 6             1580                 7.9 3650</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $og_overlap</span></span>
<span><span class="co">#&gt;                         Danio_rerio Drosophila_melanogaster Homo_sapiens</span></span>
<span><span class="co">#&gt; Danio_rerio                   13472                    5872        11365</span></span>
<span><span class="co">#&gt; Drosophila_melanogaster        5872                    6651         5866</span></span>
<span><span class="co">#&gt; Homo_sapiens                  11365                    5866        14468</span></span>
<span><span class="co">#&gt; Mus_musculus                  11345                    5863        14076</span></span>
<span><span class="co">#&gt; Takifugu_rubripes             12100                    5810        10994</span></span>
<span><span class="co">#&gt; Xenopus_tropicalis            11086                    5725        11478</span></span>
<span><span class="co">#&gt;                         Mus_musculus Takifugu_rubripes Xenopus_tropicalis</span></span>
<span><span class="co">#&gt; Danio_rerio                    11345             12100              11086</span></span>
<span><span class="co">#&gt; Drosophila_melanogaster         5863              5810               5725</span></span>
<span><span class="co">#&gt; Homo_sapiens                   14076             10994              11478</span></span>
<span><span class="co">#&gt; Mus_musculus                   14411             10976              11446</span></span>
<span><span class="co">#&gt; Takifugu_rubripes              10976             12649              10776</span></span>
<span><span class="co">#&gt; Xenopus_tropicalis             11446             10776              12302</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $duplications</span></span>
<span><span class="co">#&gt;                       Node Duplications_50</span></span>
<span><span class="co">#&gt; 1  Drosophila_melanogaster            3353</span></span>
<span><span class="co">#&gt; 2             Homo_sapiens            4527</span></span>
<span><span class="co">#&gt; 3                       N0              73</span></span>
<span><span class="co">#&gt; 4        Takifugu_rubripes            2283</span></span>
<span><span class="co">#&gt; 5             Mus_musculus            4131</span></span>
<span><span class="co">#&gt; 6              Danio_rerio            9585</span></span>
<span><span class="co">#&gt; 7                       N1            2458</span></span>
<span><span class="co">#&gt; 8                       N2            1530</span></span>
<span><span class="co">#&gt; 9                       N3             195</span></span>
<span><span class="co">#&gt; 10                      N4             745</span></span>
<span><span class="co">#&gt; 11      Xenopus_tropicalis            3650</span></span></code></pre></div>
<p>Now, we can use this list to visually explore summary statistics.</p>
<div class="section level3">
<h3 id="species-tree">Species tree<a class="anchor" aria-label="anchor" href="#species-tree"></a>
</h3>
<p>To start, one would usually want to look at the species tree to
detect possible issues that would compromise the accuracy of orthologs
detection. The tree file can be easily read with
<code><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">treeio::read.tree()</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_species_tree.html">plot_species_tree</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_species_tree-1.png" width="729.166666666667"></p>
<p>You can also include the number of gene duplications in each
node.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_species_tree.html">plot_species_tree</a></span><span class="op">(</span><span class="va">tree</span>, stats_list <span class="op">=</span> <span class="va">ortho_stats</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_species_tree_with_dups-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="species-specific-duplications">Species-specific duplications<a class="anchor" aria-label="anchor" href="#species-specific-duplications"></a>
</h3>
<p>The species tree above shows duplications per node, but it does not
show species-duplications. To visualize that, you can use the function
<code><a href="../reference/plot_duplications.html">plot_duplications()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_duplications.html">plot_duplications</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/unnamed-chunk-2-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="genes-in-orthogroups">Genes in orthogroups<a class="anchor" aria-label="anchor" href="#genes-in-orthogroups"></a>
</h3>
<p>Visualizing the percentage of genes in orthogroups is particularly
useful for quality check, since one would usually expect a large
percentage of genes in orthogroups, unless there is a very distant
species in OrthoFinder’s input proteome data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_genes_in_ogs.html">plot_genes_in_ogs</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_genes_in_ogs-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="species-specific-orthogroups">Species-specific orthogroups<a class="anchor" aria-label="anchor" href="#species-specific-orthogroups"></a>
</h3>
<p>To visualize the number of species-specific orthogroups, use the
function <code><a href="../reference/plot_species_specific_ogs.html">plot_species_specific_ogs()</a></code>. This plot can reveal
a unique gene repertoire of a particular species if it has a large
number of species-specific OGs as compared to the other ones.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_species_specific_ogs.html">plot_species_specific_ogs</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/ssOGs-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="all-in-one">All in one<a class="anchor" aria-label="anchor" href="#all-in-one"></a>
</h3>
<p>To get a complete picture of OrthoFinder results, you can combine all
plots together with <code><a href="../reference/plot_orthofinder_stats.html">plot_orthofinder_stats()</a></code>, a wrapper
that integrates all previously demonstrated plotting functions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_orthofinder_stats.html">plot_orthofinder_stats</a></span><span class="op">(</span></span>
<span>  <span class="va">tree</span>, </span>
<span>  xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  stats_list <span class="op">=</span> <span class="va">ortho_stats</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_orthofinder_stats-1.png" width="1200"></p>
</div>
<div class="section level3">
<h3 id="orthogroup-overlap">Orthogroup overlap<a class="anchor" aria-label="anchor" href="#orthogroup-overlap"></a>
</h3>
<p>You can also visualize a heatmap of pairwise orthogroup overlap
across species with <code><a href="../reference/plot_og_overlap.html">plot_og_overlap()</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_og_overlap.html">plot_og_overlap</a></span><span class="op">(</span><span class="va">ortho_stats</span><span class="op">)</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/plot_og_overlap-1.png" width="729.166666666667"></p>
</div>
<div class="section level3">
<h3 id="orthogroup-size-per-species">Orthogroup size per species<a class="anchor" aria-label="anchor" href="#orthogroup-size-per-species"></a>
</h3>
<p>If you want to take a look at the distribution of OG sizes for each
species, you can use the function <code>plot_og_sizes</code>. If you
have many extreme values and want to visualize the shape of the
distribution in a better way, you can log transform the OG sizes (with
<code>log = TRUE</code>) and/or remove OG larger than a particular
threshold (with <code>max_size = 100</code>, for example).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_og_sizes.html">plot_og_sizes</a></span><span class="op">(</span><span class="va">og</span><span class="op">)</span> </span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/og_sizes-1.png" width="729.166666666667"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_og_sizes.html">plot_og_sizes</a></span><span class="op">(</span><span class="va">og</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># natural logarithm scale</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/og_sizes-2.png" width="729.166666666667"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_og_sizes.html">plot_og_sizes</a></span><span class="op">(</span><span class="va">og</span>, max_size <span class="op">=</span> <span class="fl">100</span><span class="op">)</span> <span class="co"># only OGs with &lt;= 100 genes</span></span></code></pre></div>
<p><img src="vignette_02_assessing_orthogroup_inference_files/figure-html/og_sizes-3.png" width="729.166666666667"></p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-03-14</span></span>
<span><span class="co">#&gt;  pandoc   2.18 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package         </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  ape                5.7       <span style="color: #555555;">2023-02-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  aplot              0.1.10    <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  beeswarm           0.4.0     <span style="color: #555555;">2021-06-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics       0.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager        1.30.20   <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocStyle        * 2.24.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings         2.64.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bitops             1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bookdown           0.33      <span style="color: #555555;">2023-03-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bslib              0.4.2     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cachem             1.0.7     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cli                3.6.0     <span style="color: #555555;">2023-01-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cogeqc           * 1.3.1     <span style="color: #555555;">2023-03-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  colorspace         2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  crayon             1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  desc               1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  digest             0.6.31    <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  dplyr              1.1.0     <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  evaluate           0.20      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fansi              1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  farver             2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fastmap            1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fs                 1.6.1     <span style="color: #555555;">2023-02-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  generics           0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb       1.32.4    <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData   1.2.8     <span style="color: #555555;">2022-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ggbeeswarm         0.7.1     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggfun              0.0.9     <span style="color: #555555;">2022-11-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2            3.4.1     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggplotify          0.1.0     <span style="color: #555555;">2021-09-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggtree             3.4.4     <span style="color: #555555;">2022-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  glue               1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gridGraphics       0.5-1     <span style="color: #555555;">2020-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gtable             0.3.1     <span style="color: #555555;">2022-09-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  highr              0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmltools          0.5.4     <span style="color: #555555;">2022-12-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  igraph             1.4.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  IRanges            2.30.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  jquerylib          0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite           1.8.4     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  knitr              1.42      <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  labeling           0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lattice            0.20-45   <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  lazyeval           0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle          1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magrittr           2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  memoise            2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  munsell            0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  nlme               3.1-162   <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  patchwork          1.1.2     <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pillar             1.8.1     <span style="color: #555555;">2022-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig          2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown            2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plyr               1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  purrr              1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R6                 2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ragg               1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp               1.0.10    <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RCurl              1.98-1.10 <span style="color: #555555;">2023-01-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  reshape2           1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rlang              1.0.6     <span style="color: #555555;">2022-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown          2.20      <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot          2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  S4Vectors          0.34.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass               0.4.5     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  scales             1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo        1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringi            1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringr            1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  systemfonts        1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  textshaping        0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tibble             3.2.0     <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyr              1.3.0     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect         1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidytree           0.4.2     <span style="color: #555555;">2022-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  treeio             1.20.2    <span style="color: #555555;">2022-08-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  utf8               1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vctrs              0.5.2     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vipor              0.4.5     <span style="color: #555555;">2017-03-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  withr              2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xfun               0.37      <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XVector            0.36.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml               2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  yulab.utils        0.0.6     <span style="color: #555555;">2022-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc           1.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-emms2019orthofinder" class="csl-entry">
Emms, David M, and Steven Kelly. 2019. <span>“OrthoFinder: Phylogenetic
Orthology Inference for Comparative Genomics.”</span> <em>Genome
Biology</em> 20 (1): 1–14.
</div>
<div id="ref-li2003orthomcl" class="csl-entry">
Li, Li, Christian J Stoeckert, and David S Roos. 2003. <span>“OrthoMCL:
Identification of Ortholog Groups for Eukaryotic Genomes.”</span>
<em>Genome Research</em> 13 (9): 2178–89.
</div>
<div id="ref-van2021plaza" class="csl-entry">
Van Bel, Michiel, Francesca Silvestri, Eric M Weitz, Lukasz Kreft,
Alexander Botzki, Frederik Coppens, and Klaas Vandepoele. 2021.
<span>“PLAZA 5.0: Extending the Scope and Power of Comparative and
Functional Genomics in Plants.”</span> <em>Nucleic Acids Research</em>.
</div>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
